diff -ruN -xMakefile.in icecast2-2.3.1.orig/admin/fr_FR/listclients.xsl icecast/admin/fr_FR/listclients.xsl
--- icecast2-2.3.1.orig/admin/fr_FR/listclients.xsl	1970-01-01 01:00:00.000000000 +0100
+++ icecast/admin/fr_FR/listclients.xsl	2006-06-20 17:28:14.000000000 +0200
@@ -0,0 +1,95 @@
+<xsl:stylesheet xmlns:xsl = "http://www.w3.org/1999/XSL/Transform" version = "1.0" >
+<xsl:output omit-xml-declaration="no" method="xml" doctype-public="-//W3C//DTD XHTML 1.0 Strict//EN" doctype-system="http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd" indent="yes" encoding="UTF-8" />
+<xsl:template match = "/icestats" >
+<html>
+<head>
+<title>Icecast Streaming Media Server</title>
+<link rel="stylesheet" type="text/css" href="/style.css" />
+</head>
+<body topmargin="0" leftmargin="0" rightmargin="0" bottommargin="0">
+<img align="left" src="/icecast.png" /><h2><center>Icecast2 Admin</center></h2>
+<br />
+
+<div class="roundcont">
+<div class="roundtop">
+<img src="/corner_topleft.jpg" class="corner" style="display: none" />
+</div>
+	<table border="0" cellpadding="1" cellspacing="3">
+	<tr>        
+	    <td align="center">
+	        <a class="nav" href="stats.xsl">Admin Home</a> |
+		    <a class="nav" href="listmounts.xsl">List Mountpoints</a> | 
+        	<a class="nav" href="moveclients.xsl">Move Listeners</a> | 
+        	<a class="nav" href="/status.xsl">Index</a>
+	    </td></tr>
+	</table>
+<div class="roundbottom">
+<img src="/corner_bottomleft.jpg" class="corner" style="display: none" />
+</div>
+</div>
+<br />
+<br />
+
+
+
+<h1>Listener Stats</h1>
+<div class="roundcont">
+<div class="roundtop">
+<img src="/corner_topleft.jpg" class="corner" style="display: none" />
+</div>
+<div class="newscontent">
+<xsl:for-each select="source">
+<h3>
+
+<xsl:choose>
+<xsl:when test="authenticator">
+<a href="/auth.xsl"><img border="0" src="/key.png"/></a>
+</xsl:when>
+<xsl:otherwise>
+<a href="{@mount}.m3u"><img border="0" src="/tunein.png"/></a>
+</xsl:otherwise>
+</xsl:choose>
+<xsl:if test="server_name"><xsl:value-of select="server_name" /> </xsl:if>
+Mount Point : (<xsl:value-of select="@mount" />)
+</h3>
+	<table border="0" cellpadding="1" cellspacing="5" bgcolor="444444">
+	<tr>        
+	    <td align="center">
+			<a class="nav2" href="listclients.xsl?mount={@mount}">List Clients</a> | 
+        	<a class="nav2" href="moveclients.xsl?mount={@mount}">Move Listeners</a> | 
+			<a class="nav2" href="updatemetadata.xsl?mount={@mount}">Update Metadata</a> |
+        	<a class="nav2" href="killsource.xsl?mount={@mount}">Kill Source</a>
+	    </td></tr>
+	</table>
+<br />
+<table cellspacing="1" border="1" bordercolor="#C0C0C0" >
+		<tr>
+				<td ><center><b>IP</b></center></td>
+				<td ><center><b>Connected For</b></center></td>
+				<td ><center><b>User Agent</b></center></td>
+				<td ><center><b>Action</b></center></td>
+		</tr>
+<xsl:variable name = "themount" ><xsl:value-of select="@mount" /></xsl:variable>
+<xsl:for-each select="listener">
+		<tr>
+				<td align="center"><xsl:value-of select="IP" /><xsl:if test="username"> (<xsl:value-of select="username" />)</xsl:if></td>
+				<td align="center"><xsl:value-of select="Connected" /> seconds</td>
+				<td align="center"><xsl:value-of select="UserAgent" /></td>
+				<td align="center"><a href="killclient.xsl?mount={$themount}&amp;id={ID}">Kick</a></td>
+		</tr>
+</xsl:for-each>
+</table>
+<br />
+<br />
+</xsl:for-each>
+<xsl:text disable-output-escaping="yes">&amp;</xsl:text>nbsp;
+</div>
+<div class="roundbottom">
+<img src="/corner_bottomleft.jpg" class="corner" style="display: none" />
+</div>
+</div>
+<div class="poster">Support icecast development at <a class="nav" href="http://www.icecast.org">www.icecast.org</a></div>
+</body>
+</html>
+</xsl:template>
+</xsl:stylesheet>
diff -ruN -xMakefile.in icecast2-2.3.1.orig/admin/fr_FR/listmounts.xsl icecast/admin/fr_FR/listmounts.xsl
--- icecast2-2.3.1.orig/admin/fr_FR/listmounts.xsl	1970-01-01 01:00:00.000000000 +0100
+++ icecast/admin/fr_FR/listmounts.xsl	2006-06-20 17:28:14.000000000 +0200
@@ -0,0 +1,77 @@
+<xsl:stylesheet xmlns:xsl = "http://www.w3.org/1999/XSL/Transform" version = "1.0" >
+<xsl:output omit-xml-declaration="no" method="xml" doctype-public="-//W3C//DTD XHTML 1.0 Strict//EN" doctype-system="http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd" indent="yes" encoding="UTF-8" />
+<xsl:template match = "/icestats" >
+<html>
+<head>
+<title>Icecast Streaming Media Server</title>
+<link rel="stylesheet" type="text/css" href="/style.css" />
+</head>
+<body topmargin="0" leftmargin="0" rightmargin="0" bottommargin="0">
+<img align="left" src="/icecast.png" /><h2><center>Icecast2 Admin</center></h2>
+<br />
+
+<div class="roundcont">
+<div class="roundtop">
+<img src="/corner_topleft.jpg" class="corner" style="display: none" />
+</div>
+	<table border="0" cellpadding="1" cellspacing="3">
+	<tr>        
+	    <td align="center">
+	        <a class="nav" href="stats.xsl">Admin Home</a> |
+		    <a class="nav" href="listmounts.xsl">List Mountpoints</a> | 
+        	<a class="nav" href="moveclients.xsl">Move Listeners</a> | 
+        	<a class="nav" href="/status.xsl">Index</a>
+	    </td></tr>
+	</table>
+<div class="roundbottom">
+<img src="/corner_bottomleft.jpg" class="corner" style="display: none" />
+</div>
+</div>
+<br />
+<br />
+
+
+<h1>Active Mountpoints</h1>
+<div class="roundcont">
+<div class="roundtop">
+<img src="/corner_topleft.jpg" class="corner" style="display: none" />
+</div>
+<div class="newscontent">
+<xsl:for-each select="source">
+<h3>
+<xsl:choose>
+<xsl:when test="authenticator">
+<a href="/auth.xsl"><img border="0" src="/key.png"/></a>
+</xsl:when>
+<xsl:otherwise>
+<a href="{@mount}.m3u"><img border="0" src="/tunein.png"/></a>
+</xsl:otherwise>
+</xsl:choose>
+<xsl:if test="server_name"><xsl:value-of select="server_name" /> </xsl:if>
+Mount Point : (<xsl:value-of select="@mount" />)
+</h3>
+	<table border="0" cellpadding="1" cellspacing="5" bgcolor="444444">
+	<tr>        
+	    <td align="center">
+			<a class="nav2" href="listclients.xsl?mount={@mount}">List Clients</a> | 
+        	<a class="nav2" href="moveclients.xsl?mount={@mount}">Move Listeners</a> | 
+			<a class="nav2" href="updatemetadata.xsl?mount={@mount}">Update Metadata</a> |
+        	<a class="nav2" href="killsource.xsl?mount={@mount}">Kill Source</a>
+                <xsl:if test="authenticator"> | <a class="nav2" href="manageauth.xsl?mount={@mount}">Manage Authentication</a></xsl:if>
+	    </td></tr>
+	</table>
+<br />
+<p><xsl:value-of select="listeners" /> Listener(s)</p>
+<br></br>
+</xsl:for-each>
+<xsl:text disable-output-escaping="yes">&amp;</xsl:text>nbsp;
+</div>
+<div class="roundbottom">
+<img src="/corner_bottomleft.jpg" class="corner" style="display: none" />
+</div>
+</div>
+<div class="poster">Support icecast development at <a class="nav" href="http://www.icecast.org">www.icecast.org</a></div>
+</body>
+</html>
+</xsl:template>
+</xsl:stylesheet>
diff -ruN -xMakefile.in icecast2-2.3.1.orig/admin/fr_FR/Makefile.am icecast/admin/fr_FR/Makefile.am
--- icecast2-2.3.1.orig/admin/fr_FR/Makefile.am	1970-01-01 01:00:00.000000000 +0100
+++ icecast/admin/fr_FR/Makefile.am	2006-06-20 17:28:14.000000000 +0200
@@ -0,0 +1,8 @@
+## Process this file with automake to produce Makefile.in
+
+AUTOMAKE_OPTIONS = foreign
+
+admindir = $(pkgdatadir)/admin
+dist_admin_DATA = listclients.xsl listmounts.xsl moveclients.xsl response.xsl \
+	stats.xsl manageauth.xsl updatemetadata.xsl
+
diff -ruN -xMakefile.in icecast2-2.3.1.orig/admin/fr_FR/manageauth.xsl icecast/admin/fr_FR/manageauth.xsl
--- icecast2-2.3.1.orig/admin/fr_FR/manageauth.xsl	1970-01-01 01:00:00.000000000 +0100
+++ icecast/admin/fr_FR/manageauth.xsl	2006-06-20 17:28:14.000000000 +0200
@@ -0,0 +1,95 @@
+<xsl:stylesheet xmlns:xsl = "http://www.w3.org/1999/XSL/Transform" version = "1.0" >
+<xsl:output omit-xml-declaration="no" method="xml" doctype-public="-//W3C//DTD XHTML 1.0 Strict//EN" doctype-system="http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd" indent="yes" encoding="UTF-8" />
+<xsl:template match = "/icestats" >
+<html>
+<head>
+<title>Icecast Streaming Media Server</title>
+<link rel="stylesheet" type="text/css" href="/style.css" />
+</head>
+<body topmargin="0" leftmargin="0" rightmargin="0" bottommargin="0">
+<img align="left" src="/icecast.png" /><h2><center>Icecast2 Admin</center></h2>
+<br />
+
+<div class="roundcont">
+<div class="roundtop">
+<img src="/corner_topleft.jpg" class="corner" style="display: none" />
+</div>
+	<table border="0" cellpadding="1" cellspacing="3">
+	<tr>        
+	    <td align="center">
+	        <a class="nav" href="stats.xsl">Admin Home</a> |
+		    <a class="nav" href="listmounts.xsl">List Mountpoints</a> | 
+        	<a class="nav" href="moveclients.xsl">Move Listeners</a> | 
+        	<a class="nav" href="/status.xsl">Index</a>
+	    </td></tr>
+	</table>
+<div class="roundbottom">
+<img src="/corner_bottomleft.jpg" class="corner" style="display: none" />
+</div>
+</div>
+<br />
+<br />
+
+<h2>Show defined users</h2>
+<div class="roundcont">
+<div class="roundtop">
+<img src="/corner_topleft.jpg" class="corner" style="display: none" />
+</div>
+<div class="newscontent">
+<xsl:for-each select="iceresponse">
+<xsl:value-of select="message" /> 
+</xsl:for-each>
+<xsl:for-each select="source">
+<h3>
+<xsl:if test="server_name"><xsl:value-of select="server_name" /> </xsl:if>
+(<xsl:value-of select="@mount" />)</h3>
+	<table border="0" cellpadding="1" cellspacing="5" bgcolor="444444">
+	<tr>        
+	    <td align="center">
+			<a class="nav2" href="listclients.xsl?mount={@mount}">List Clients</a> |
+        	<a class="nav2" href="moveclients.xsl?mount={@mount}">Move Listeners</a> | 
+			<a class="nav2" href="updatemetadata.xsl?mount={@mount}">Update Metadata</a> |
+        	<a class="nav2" href="killsource.xsl?mount={@mount}">Kill Source</a>
+	    </td></tr>
+	</table>
+<br></br>
+<form method="GET" action="manageauth.xsl">
+<table cellpadding="2" cellspacing="4" border="0" >
+		<tr>
+				<td ><b>User Id</b></td>
+				<td ><b>Password</b></td>
+				<td ></td>
+		</tr>
+<xsl:variable name = "themount" ><xsl:value-of select="@mount" /></xsl:variable>
+<xsl:for-each select="User">
+		<tr>
+				<td><xsl:value-of select="username" /></td>
+				<td><xsl:value-of select="password" /></td>
+				<td><a class="nav2" href="manageauth.xsl?mount={$themount}&amp;username={username}&amp;action=delete">delete</a></td>
+		</tr>
+</xsl:for-each>
+		<tr>
+				<td ><input type="text" name="username" /></td>
+				<td ><input type="text" name="password" /></td>
+		</tr>
+		<tr>
+				<td colspan="2"><input type="Submit" name="Submit" value="Add New User" /></td>
+		</tr>
+</table>
+<input type="hidden" name="mount" value="{@mount}"/>
+<input type="hidden" name="action" value="add"/>
+</form>
+<br />
+<br />
+</xsl:for-each>
+<xsl:text disable-output-escaping="yes">&amp;</xsl:text>nbsp;
+</div>
+<div class="roundbottom">
+<img src="/corner_bottomleft.jpg" class="corner" style="display: none" />
+</div>
+</div>
+<div class="poster">Support icecast development at <a class="nav" href="http://www.icecast.org">www.icecast.org</a></div>
+</body>
+</html>
+</xsl:template>
+</xsl:stylesheet>
diff -ruN -xMakefile.in icecast2-2.3.1.orig/admin/fr_FR/moveclients.xsl icecast/admin/fr_FR/moveclients.xsl
--- icecast2-2.3.1.orig/admin/fr_FR/moveclients.xsl	1970-01-01 01:00:00.000000000 +0100
+++ icecast/admin/fr_FR/moveclients.xsl	2006-06-20 17:28:14.000000000 +0200
@@ -0,0 +1,63 @@
+<xsl:stylesheet xmlns:xsl = "http://www.w3.org/1999/XSL/Transform" version = "1.0" >
+<xsl:output omit-xml-declaration="no" method="xml" doctype-public="-//W3C//DTD XHTML 1.0 Strict//EN" doctype-system="http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd" indent="yes" encoding="UTF-8" />
+<xsl:template match = "/icestats" >
+<html>
+<head>
+<title>Icecast Streaming Media Server</title>
+<link rel="stylesheet" type="text/css" href="/style.css" />
+</head>
+<body topmargin="0" leftmargin="0" rightmargin="0" bottommargin="0">
+<img align="left" src="/icecast.png" /><h2><center>Icecast2 Admin</center></h2>
+<br />
+
+<div class="roundcont">
+<div class="roundtop">
+<img src="/corner_topleft.jpg" class="corner" style="display: none" />
+</div>
+	<table border="0" cellpadding="1" cellspacing="3">
+	<tr>        
+	    <td align="center">
+	        <a class="nav" href="stats.xsl">Admin Home</a> |
+		    <a class="nav" href="listmounts.xsl">List Mountpoints</a> | 
+        	<a class="nav" href="moveclients.xsl">Move Listeners</a> | 
+        	<a class="nav" href="/status.xsl">Index</a>
+	    </td></tr>
+	</table>
+<div class="roundbottom">
+<img src="/corner_bottomleft.jpg" class="corner" style="display: none" />
+</div>
+</div>
+<br />
+<br />
+
+<xsl:variable name = "currentmount" ><xsl:value-of select="current_source" /></xsl:variable>
+<h1>Moving Listeners From (<xsl:value-of select="current_source" />)</h1>
+<div class="roundcont">
+<div class="roundtop">
+<img src="/corner_topleft.jpg" class="corner" style="display: none" />
+</div>
+<div class="newscontent">
+<h3>Move to which mountpoint ?</h3>
+<xsl:for-each select="source">
+	<table border="0" cellpadding="1" cellspacing="5" >
+	<tr>        
+		<td>Move from (<xsl:copy-of select="$currentmount" />) to (<xsl:value-of select="@mount" />)</td>
+		<td><xsl:value-of select="listeners" /> Listeners</td>
+		<td><a class="nav2" href="moveclients.xsl?mount={$currentmount}&amp;destination={@mount}">Move Clients</a></td>
+	</tr>        
+	</table>
+<br />
+<br />
+</xsl:for-each>
+<xsl:text disable-output-escaping="yes">&amp;</xsl:text>nbsp;
+</div>
+<div class="roundbottom">
+<img src="/corner_bottomleft.jpg" class="corner" style="display: none" />
+</div>
+</div>
+<div class="poster">Support icecast development at <a class="nav" href="http://www.icecast.org">www.icecast.org</a></div>
+</body>
+</html>
+
+</xsl:template>
+</xsl:stylesheet>
diff -ruN -xMakefile.in icecast2-2.3.1.orig/admin/fr_FR/response.xsl icecast/admin/fr_FR/response.xsl
--- icecast2-2.3.1.orig/admin/fr_FR/response.xsl	1970-01-01 01:00:00.000000000 +0100
+++ icecast/admin/fr_FR/response.xsl	2006-06-20 17:28:14.000000000 +0200
@@ -0,0 +1,56 @@
+<xsl:stylesheet xmlns:xsl = "http://www.w3.org/1999/XSL/Transform" version = "1.0" >
+<xsl:output omit-xml-declaration="no" method="xml" doctype-public="-//W3C//DTD XHTML 1.0 Strict//EN" doctype-system="http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd" indent="yes" encoding="UTF-8" />
+<xsl:template match = "/iceresponse" >
+<html>
+<head>
+<title>Icecast Streaming Media Server</title>
+<link rel="stylesheet" type="text/css" href="/style.css" />
+</head>
+<body topmargin="0" leftmargin="0" rightmargin="0" bottommargin="0">
+<img align="left" src="/icecast.png" /><h2><center>Icecast2 Admin</center></h2>
+<br />
+
+<div class="roundcont">
+<div class="roundtop">
+<img src="/corner_topleft.jpg" class="corner" style="display: none" />
+</div>
+	<table border="0" cellpadding="1" cellspacing="3">
+	<tr>        
+	    <td align="center">
+	        <a class="nav" href="stats.xsl">Admin Home</a> |
+		    <a class="nav" href="listmounts.xsl">List Mountpoints</a> | 
+        	<a class="nav" href="moveclients.xsl">Move Listeners</a> | 
+        	<a class="nav" href="/status.xsl">Index</a>
+	    </td></tr>
+	</table>
+<div class="roundbottom">
+<img src="/corner_bottomleft.jpg" class="corner" style="display: none" />
+</div>
+</div>
+<br />
+<br />
+
+<h1>Icecast Server Response</h1>
+<div class="roundcont">
+<div class="roundtop">
+<img src="/corner_topleft.jpg" class="corner" style="display: none" />
+</div>
+<div class="newscontent">
+<h3>Response</h3>
+<xsl:for-each select="/iceresponse">
+Message : <xsl:value-of select="message" /><br></br>
+Return Code: <xsl:value-of select="return" /><br></br>
+</xsl:for-each>
+<br />
+<br />
+</div>
+<div class="roundbottom">
+<img src="/corner_bottomleft.jpg" class="corner" style="display: none" />
+</div>
+</div>
+<div class="poster">Support icecast development at <a class="nav" href="http://www.icecast.org">www.icecast.org</a></div>
+</body>
+</html>
+
+</xsl:template>
+</xsl:stylesheet>
diff -ruN -xMakefile.in icecast2-2.3.1.orig/admin/fr_FR/stats.xsl icecast/admin/fr_FR/stats.xsl
--- icecast2-2.3.1.orig/admin/fr_FR/stats.xsl	1970-01-01 01:00:00.000000000 +0100
+++ icecast/admin/fr_FR/stats.xsl	2006-06-20 17:28:14.000000000 +0200
@@ -0,0 +1,117 @@
+<xsl:stylesheet xmlns:xsl = "http://www.w3.org/1999/XSL/Transform" version = "1.0" >
+<xsl:output omit-xml-declaration="no" method="xml" doctype-public="-//W3C//DTD XHTML 1.0 Strict//EN" doctype-system="http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd" indent="yes" encoding="UTF-8" />
+<xsl:template match = "/icestats" >
+<html>
+<head>
+<title>Icecast Streaming Media Server</title>
+<link rel="stylesheet" type="text/css" href="/style.css" />
+</head>
+<body topmargin="0" leftmargin="0" rightmargin="0" bottommargin="0">
+<img align="left" src="/icecast.png" /><h2><center>Icecast2 Admin</center></h2>
+<br />
+<!--header menu -->
+<div class="roundcont">
+<div class="roundtop">
+<img src="/corner_topleft.jpg" class="corner" style="display: none" />
+</div>
+	<table border="0" cellpadding="1" cellspacing="3">
+	<tr>        
+	    <td align="center">
+	        <a class="nav" href="stats.xsl">Admin Home</a> | 
+		    <a class="nav" href="listmounts.xsl">List Mountpoints</a> | 
+        	<a class="nav" href="moveclients.xsl">Move Listeners</a> | 
+        	<a class="nav" href="/status.xsl">Index</a>
+	    </td></tr>
+	</table>
+<div class="roundbottom">
+<img src="/corner_bottomleft.jpg" class="corner" style="display: none" />
+</div>
+</div>
+<br />
+<br />
+<!--end header menu -->
+
+<!--global server stats-->
+<div class="roundcont">
+<div class="roundtop">
+<img src="/corner_topleft.jpg" class="corner" style="display: none" />
+</div>
+<div class="newscontent">
+<h3>Global Server Stats</h3>
+<table border="0" cellpadding="4">
+<xsl:for-each select="/icestats">
+<xsl:for-each select="*">
+<xsl:if test = "name()!='source'"> 
+<tr>
+	<td width="130"><xsl:value-of select="name()" /></td>
+	<td class="streamdata"><xsl:value-of select="." /></td>
+</tr>
+</xsl:if>
+</xsl:for-each>
+</xsl:for-each>
+</table>
+</div>
+<div class="roundbottom">
+<img src="/corner_bottomleft.jpg" class="corner" style="display: none" />
+</div>
+</div>
+<br />
+<br />
+<!--end global server stats-->
+
+<!--mount point stats-->
+<xsl:for-each select="source">
+<xsl:if test = "listeners!=''"> 
+<div class="roundcont">
+<div class="roundtop">
+<img src="/corner_topleft.jpg" class="corner" style="display: none" />
+</div>
+<div class="newscontent">
+<h3>
+<xsl:choose>
+<xsl:when test="authenticator">
+<a href="/auth.xsl"><img border="0" src="/key.png"/></a>
+</xsl:when>
+<xsl:otherwise>
+<a href="{@mount}.m3u"><img border="0" src="/tunein.png"/></a>
+</xsl:otherwise>
+</xsl:choose>
+Mount Point : (<xsl:value-of select="@mount" />)
+</h3>
+	<table border="0" cellpadding="1" cellspacing="5" bgcolor="444444">
+	<tr>        
+	    <td align="center">
+		    <a class="nav2" href="listclients.xsl?mount={@mount}">List Clients</a> | 
+        	<a class="nav2" href="moveclients.xsl?mount={@mount}">Move MountPoints</a> | 
+        	<a class="nav2" href="updatemetadata.xsl?mount={@mount}">Update Metadata</a> |
+        	<a class="nav2" href="killsource.xsl?mount={@mount}">Kill Source</a>
+                <xsl:if test="authenticator"> | <a class="nav2" href="manageauth.xsl?mount={@mount}">Manage Authentication</a></xsl:if>
+	    </td></tr>
+	</table>
+<br />
+<table cellpadding="5" cellspacing="0" border="0">
+	<xsl:for-each select="*">
+	<tr>
+		<td width="130"><xsl:value-of select="name()" /></td>
+		<td class="streamdata"><xsl:value-of select="." /></td>
+	</tr>
+	</xsl:for-each>
+</table>
+</div>
+<div class="roundbottom">
+<img src="/corner_bottomleft.jpg" class="corner" style="display: none" />
+</div>
+</div>
+<br />
+<br />
+</xsl:if>
+</xsl:for-each>
+<xsl:text disable-output-escaping="yes">&amp;</xsl:text>nbsp;
+
+
+<!--end mount point stats-->
+<div class="poster">Support icecast development at <a class="nav" href="http://www.icecast.org">www.icecast.org</a></div>
+</body>
+</html>
+</xsl:template>
+</xsl:stylesheet>
diff -ruN -xMakefile.in icecast2-2.3.1.orig/admin/fr_FR/updatemetadata.xsl icecast/admin/fr_FR/updatemetadata.xsl
--- icecast2-2.3.1.orig/admin/fr_FR/updatemetadata.xsl	1970-01-01 01:00:00.000000000 +0100
+++ icecast/admin/fr_FR/updatemetadata.xsl	2006-06-20 17:28:14.000000000 +0200
@@ -0,0 +1,66 @@
+<xsl:stylesheet xmlns:xsl = "http://www.w3.org/1999/XSL/Transform" version = "1.0" >
+<xsl:output omit-xml-declaration="no" method="xml" doctype-public="-//W3C//DTD XHTML 1.0 Strict//EN" doctype-system="http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd" indent="yes" encoding="UTF-8" />
+<xsl:template match = "/icestats" >
+<html>
+<head>
+<title>Icecast Streaming Media Server</title>
+<link rel="stylesheet" type="text/css" href="/style.css" />
+</head>
+<body topmargin="0" leftmargin="0" rightmargin="0" bottommargin="0">
+<img align="left" src="/icecast.png" /><h2><center>Icecast2 Admin</center></h2>
+<br />
+
+<div class="roundcont">
+<div class="roundtop">
+<img src="/corner_topleft.jpg" class="corner" style="display: none" />
+</div>
+	<table border="0" cellpadding="1" cellspacing="3">
+	<tr>        
+	    <td align="center">
+	        <a class="nav" href="stats.xsl">Admin Home</a> |
+		    <a class="nav" href="listmounts.xsl">List Mountpoints</a> | 
+        	<a class="nav" href="moveclients.xsl">Move Listeners</a> | 
+        	<a class="nav" href="/status.xsl">Index</a>
+	    </td></tr>
+	</table>
+<div class="roundbottom">
+<img src="/corner_bottomleft.jpg" class="corner" style="display: none" />
+</div>
+</div>
+<br />
+<br />
+
+<h1>Update Metadata</h1>
+<div class="roundcont">
+<div class="roundtop">
+<img src="/corner_topleft.jpg" class="corner" style="display: none" />
+</div>
+<div class="newscontent">
+<xsl:for-each select="source">
+<h3>
+<xsl:if test="server_name"><xsl:value-of select="server_name" /> </xsl:if>
+(<xsl:value-of select="@mount" />)</h3>
+
+<form method="GET" action="/admin/metadata.xsl">
+<table border="0" cellpadding="4">
+<tr><td>Metadata : <input type="text" name="song"/></td></tr>
+<tr><td><input type="Submit" value="Update"/></td></tr>
+</table>
+<input type="hidden" name="mount" value="{@mount}"/>
+<input type="hidden" name="mode" value="updinfo"/>
+</form>
+
+<br />
+<br />
+</xsl:for-each>
+<xsl:text disable-output-escaping="yes">&amp;</xsl:text>nbsp;
+</div>
+<div class="roundbottom">
+<img src="/corner_bottomleft.jpg" class="corner" style="display: none" />
+</div>
+</div>
+<div class="poster">Support icecast development at <a class="nav" href="http://www.icecast.org">www.icecast.org</a></div>
+</body>
+</html>
+</xsl:template>
+</xsl:stylesheet>
diff -ruN -xMakefile.in icecast2-2.3.1.orig/autogen.sh icecast/autogen.sh
--- icecast2-2.3.1.orig/autogen.sh	1970-01-01 01:00:00.000000000 +0100
+++ icecast/autogen.sh	2006-06-20 17:28:20.000000000 +0200
@@ -0,0 +1,113 @@
+#!/bin/sh
+# Run this to set up the build system: configure, makefiles, etc.
+# (based on the version in enlightenment's cvs)
+
+package="icecast"
+
+olddir=`pwd`
+srcdir=`dirname $0`
+test -z "$srcdir" && srcdir=.
+
+cd "$srcdir"
+DIE=0
+
+(autoconf --version) < /dev/null > /dev/null 2>&1 || {
+        echo
+        echo "You must have autoconf installed to compile $package."
+        echo "Download the appropriate package for your distribution,"
+        echo "or get the source tarball at ftp://ftp.gnu.org/pub/gnu/"
+        DIE=1
+}
+VERSIONGREP="sed -e s/.*[^0-9\.]\([0-9]\.[0-9]\).*/\1/"
+                                                                                
+# do we need automake?
+if test -r Makefile.am; then
+    echo Checking for automake version
+    options=`fgrep AUTOMAKE_OPTIONS Makefile.am`
+    AM_NEEDED=`echo "$options" | $VERSIONGREP`
+    AM_PROGS=automake
+    AC_PROGS=aclocal
+    if test -n "$AM_NEEDED" && test "x$AM_NEEDED" != "x$options"
+    then
+        AM_PROGS="automake-$AM_NEEDED automake$AM_NEEDED $AM_PROGS"
+        AC_PROGS="aclocal-$AM_NEEDED aclocal$AM_NEEDED $AC_PROGS"
+    else
+        AM_NEEDED=""
+    fi
+    AM_PROGS="$AUTOMAKE $AM_PROGS"
+    AC_PROGS="$ACLOCAL $AC_PROGS"
+    for am in $AM_PROGS; do
+      ($am --version > /dev/null 2>&1) 2>/dev/null || continue
+      ver=`$am --version | head -1 | $VERSIONGREP`
+      AWK_RES=`echo $ver $AM_NEEDED | awk '{ if ( $1 >= $2 ) print "yes"; else print "no" }'`
+      if test "$AWK_RES" = "yes"; then
+        AUTOMAKE=$am
+        echo "  found $AUTOMAKE"
+        break
+      fi
+    done
+    for ac in $AC_PROGS; do
+      ($ac --version > /dev/null 2>&1) 2>/dev/null || continue
+      ver=`$ac --version < /dev/null | head -1 | $VERSIONGREP`
+      AWK_RES=`echo $ver $AM_NEEDED | awk '{ if ( $1 >= $2 ) print "yes"; else print "no" }'`
+      if test "$AWK_RES" = "yes"; then
+        ACLOCAL=$ac
+        echo "  found $ACLOCAL"
+        break
+      fi
+    done
+    test -z $AUTOMAKE || test -z $ACLOCAL && {
+        echo
+        if test -n "$AM_NEEDED"; then
+            echo "You must have automake version $AM_NEEDED installed"
+            echo "to compile $package."
+        else
+            echo "You must have automake installed to compile $package."
+        fi
+        echo "Download the appropriate package for your distribution,"
+        echo "or get the source tarball at ftp://ftp.gnu.org/pub/gnu/"
+        DIE=1
+      }
+fi
+
+(libtoolize --version)  > /dev/null 2>&1 || {
+	echo
+	echo "You must have libtool installed to compile $package."
+	echo "Download the appropriate package for your system,"
+	echo "or get the source from one of the GNU ftp sites"
+	echo "listed in http://www.gnu.org/order/ftp.html"
+	DIE=1
+}
+
+if test "$DIE" -eq 1; then
+        exit 1
+fi
+
+echo "Generating configuration files for $package, please wait...."
+
+ACLOCAL_FLAGS="$ACLOCAL_FLAGS -I m4"
+if test -n "$ACLOCAL"; then
+  echo "  $ACLOCAL $ACLOCAL_FLAGS"
+  $ACLOCAL $ACLOCAL_FLAGS
+fi
+
+echo "  autoheader"
+autoheader
+
+echo "  libtoolize --automake"
+libtoolize --automake
+
+if test -n "$AUTOMAKE"; then
+  echo "  $AUTOMAKE --add-missing"
+  $AUTOMAKE --add-missing 
+fi
+
+echo "  autoconf"
+autoconf
+
+if test -z "$*"; then
+        echo "I am going to run ./configure with no arguments - if you wish "
+        echo "to pass any to it, please specify them on the $0 command line."
+fi
+cd $olddir
+$srcdir/configure "$@" && echo
diff -ruN -xMakefile.in icecast2-2.3.1.orig/conf/icecast.xml.in icecast/conf/icecast.xml.in
--- icecast2-2.3.1.orig/conf/icecast.xml.in	2005-10-24 16:09:51.000000000 +0200
+++ icecast/conf/icecast.xml.in	2006-06-20 17:28:12.000000000 +0200
@@ -37,10 +37,6 @@
         <yp-url-timeout>15</yp-url-timeout>
         <yp-url>http://dir.xiph.org/cgi-bin/yp-cgi</yp-url>
     </directory>
-    <directory>
-        <yp-url-timeout>15</yp-url-timeout>
-        <yp-url>http://www.oddsock.org/cgi-bin/yp-cgi</yp-url>
-    </directory>
      -->
 
     <!-- This is the hostname other people will use to connect to your server.
diff -ruN -xMakefile.in icecast2-2.3.1.orig/m4/ac_config_libconfig_in.m4 icecast/m4/ac_config_libconfig_in.m4
--- icecast2-2.3.1.orig/m4/ac_config_libconfig_in.m4	1970-01-01 01:00:00.000000000 +0100
+++ icecast/m4/ac_config_libconfig_in.m4	2006-06-20 17:28:32.000000000 +0200
@@ -0,0 +1,270 @@
+dnl Available from the GNU Autoconf Macro Archive at:
+dnl http://www.gnu.org/software/ac-archive/htmldoc/ac_config_libconfig_in.html
+dnl
+# AC_CONFIG_LIBCONFIG_IN(LIBRARY, DESTINATION, MODULES)
+# -----------------------------------------------------
+# Generate a custom LIBRARY-config script.  Create the script in the
+# DESTINATION directory, including support for MODULES.
+AC_DEFUN([AC_CONFIG_LIBCONFIG_IN],
+[# create a custom library-config file ($1-config)
+m4_if(AC_CONFIG_LIBCONFIG_IN_USEPKGCONFIG, [true],
+      [AC_PATH_PROG(PKG_CONFIG, pkg-config)])
+pushdef([LIBCONFIG_DIR], [m4_if([$2], , , [$2/])])
+LIBCONFIG_FILE="LIBCONFIG_DIR[]$1-config.in"
+AC_SUBST(target)dnl
+AC_SUBST(host)dnl
+AC_SUBST(build)dnl
+dnl create directory if it does not preexist
+m4_if([$2], , , [AS_MKDIR_P([$2])])
+AC_MSG_NOTICE([creating $LIBCONFIG_FILE])
+echo '#! /bin/sh' >$LIBCONFIG_FILE
+echo "# $1-config library configuration script" >>$LIBCONFIG_FILE
+echo '# generated by ac_config_libconfig_in.m4' >>$LIBCONFIG_FILE
+echo ' ' >>$LIBCONFIG_FILE
+echo 'template_version="1.0.0"' >>$LIBCONFIG_FILE
+echo ' ' >>$LIBCONFIG_FILE
+echo 'package="@PACKAGE@"' >>$LIBCONFIG_FILE
+echo ' ' >>$LIBCONFIG_FILE
+echo '# usage instructions if no options given' >>$LIBCONFIG_FILE
+echo 'if test "'"\$""#"'" -eq 0; then' >>$LIBCONFIG_FILE
+echo '   cat <<EOF' >>$LIBCONFIG_FILE
+m4_if($3, ,
+      [echo 'Usage: $1-config [[OPTIONS]]' >>$LIBCONFIG_FILE],
+      [echo 'Usage: $1-config [[OPTIONS]] [[LIBRARIES]]' >>$LIBCONFIG_FILE])
+echo 'Options:' >>$LIBCONFIG_FILE
+echo '        [[--prefix[=DIR]]]' >>$LIBCONFIG_FILE
+echo '        [[--exec-prefix[=DIR]]]' >>$LIBCONFIG_FILE
+echo '        [[--package]]' >>$LIBCONFIG_FILE
+echo '        [[--version]]' >>$LIBCONFIG_FILE
+echo '        [[--cppflags]]' >>$LIBCONFIG_FILE
+echo '        [[--cflags-only]]' >>$LIBCONFIG_FILE
+echo '        [[--cflags]]' >>$LIBCONFIG_FILE
+echo '        [[--libs]]' >>$LIBCONFIG_FILE
+echo '        [[--help]]' >>$LIBCONFIG_FILE
+m4_if($3, , ,
+      [echo 'Libraries:' >>$LIBCONFIG_FILE
+       for module in $1 $3 ; do
+         echo "        $module" >>$LIBCONFIG_FILE ;
+       done])
+echo 'EOF' >>$LIBCONFIG_FILE
+echo 'fi' >>$LIBCONFIG_FILE
+echo ' ' >>$LIBCONFIG_FILE
+echo '# parse options' >>$LIBCONFIG_FILE
+echo 'o=""' >>$LIBCONFIG_FILE
+echo 'h=""' >>$LIBCONFIG_FILE
+echo 'for i' >>$LIBCONFIG_FILE
+echo 'do' >>$LIBCONFIG_FILE
+echo '  case $i in' >>$LIBCONFIG_FILE
+options="prefix exec-prefix eprefix package version cppflags cflags-only cflags libs bindir sbindir libexecdir datadir sysconfdir sharedstatedir localstatedir libdir infodir mandir target host build pkgdatadir pkglibdir pkgincludedir template-version help"
+echo '    --prefix=*) prefix=`echo $i | sed -e "s/--prefix=//"` ;;' >>$LIBCONFIG_FILE
+echo '    --exec-prefix=*) exec_prefix=`echo $i | sed -e "s/--exec-prefix=//"` ;;' >>$LIBCONFIG_FILE
+echo '    --eprefix=*) exec_prefix=`echo $i | sed -e "s/--eprefix=//"` ;;' >>$LIBCONFIG_FILE
+for option in $options ; do
+  case $option in
+    exec-prefix)  echo "    --$option) echo_exec_prefix=\"yes\" ;;" >>$LIBCONFIG_FILE ;;
+    template-version)  echo "    --$option) echo_template_version=\"yes\" ;;" >>$LIBCONFIG_FILE ;;
+    cflags-only)  echo "    --$option) echo_cflags_only=\"yes\" ;;" >>$LIBCONFIG_FILE ;;
+    *)  echo "    --$option) echo_$option=\"yes\" ;;" >>$LIBCONFIG_FILE ;;
+  esac
+done
+m4_if($3, , ,
+      [for module in $1 $3 ; do
+         echo "  $module) echo_module_$module=\"yes\" ;" >>$LIBCONFIG_FILE ;
+         echo '    echo_module="yes" ;;' >>$LIBCONFIG_FILE ;
+       done])
+echo '    //*|/*//*|./*//*)        echo_extra="yes" ;;' >>$LIBCONFIG_FILE
+echo '    *) eval "echo Unknown option: $i" ; exit 1 ;;' >>$LIBCONFIG_FILE
+echo '  esac' >>$LIBCONFIG_FILE
+echo 'done' >>$LIBCONFIG_FILE
+echo ' ' >>$LIBCONFIG_FILE
+# in the order of occurence a standard automake Makefile
+echo '# defaults from configure; set only if not set previously' >>$LIBCONFIG_FILE
+vars="prefix exec_prefix bindir sbindir libexecdir datadir sysconfdir sharedstatedir localstatedir libdir infodir mandir includedir target host build"
+for var in $vars ; do
+  echo "if test -z \"\$$var\" ; then" >>$LIBCONFIG_FILE
+  echo "  $var=\"@$var@\"" >>$LIBCONFIG_FILE
+  echo 'fi' >>$LIBCONFIG_FILE
+done
+echo ' ' >>$LIBCONFIG_FILE
+echo '# generate output' >>$LIBCONFIG_FILE
+echo 'if test x$echo_module != xyes ; then' >>$LIBCONFIG_FILE
+echo '  echo_module_$1="yes"' >>$LIBCONFIG_FILE
+echo 'fi' >>$LIBCONFIG_FILE
+AC_CONFIG_LIBCONFIG_IN_MODULES(m4_if([$3], , [$1], [m4_translit([$1 $3], [ ], [,])]))dnl
+for option in $options extra; do
+  case $option in
+    exec-prefix)  echo "if test x\$echo_exec_prefix = xyes ; then" >>$LIBCONFIG_FILE ;;
+    template-version)  echo "if test x\$echo_template_version = xyes ; then" >>$LIBCONFIG_FILE ;;
+    cflags-only)  echo "if test x\$echo_cflags_only = xyes ; then" >>$LIBCONFIG_FILE ;;
+    *)  echo "if test x\$echo_$option = xyes ; then" >>$LIBCONFIG_FILE ;;
+  esac
+  case $option in
+    exec-prefix | eprefix)  echo '  o="$o $exec_prefix"' >>$LIBCONFIG_FILE ;;
+    template-version)  echo '  o="$o $template_version"' >>$LIBCONFIG_FILE ;;
+    cflags-only)  echo '  o="$o $cflags"' >>$LIBCONFIG_FILE ;;
+    cppflags)  echo '  o="$o $cppflags"' >>$LIBCONFIG_FILE ;;
+    cflags)
+#      echo '  i=`eval echo "$includedir"`' >>$LIBCONFIG_FILE ;
+#      echo '  i=`eval echo "$i"`' >>$LIBCONFIG_FILE ;
+#      echo '  if test "_$i" != "_/usr/include" ; then' >>$LIBCONFIG_FILE ;
+#      echo '    o="$o -I$includedir"' >>$LIBCONFIG_FILE ;
+#      echo '  fi' >>$LIBCONFIG_FILE ;
+      echo '  o="$o $cppflags $cflags"' >>$LIBCONFIG_FILE ;;
+    libs)  echo '  o="$o -L$libdir $libs"' >>$LIBCONFIG_FILE ;;
+    help)  echo '  h="1"' >>$LIBCONFIG_FILE ;;
+    pkgdatadir)  echo "  o=\"$o \${datadir}/\${package}\"" >>$LIBCONFIG_FILE ;;
+    pkglibdir)  echo "  o=\"$o \${libdir}/\${package}\"" >>$LIBCONFIG_FILE ;;
+    pkgincludedir)  echo "  o=\"$o \${includedir}/\${package}\"" >>$LIBCONFIG_FILE ;;
+    extra)
+      echo '  v=`echo $i | sed -e s://:\$:g`' >>$LIBCONFIG_FILE ;
+      echo '  v=`eval "echo $v"`' >>$LIBCONFIG_FILE ;
+      echo '  o="$o $v"' >>$LIBCONFIG_FILE ;;
+    *)  echo "  o=\"$o \$$option\"" >>$LIBCONFIG_FILE
+  esac
+  echo 'fi' >>$LIBCONFIG_FILE
+done
+echo ' ' >>$LIBCONFIG_FILE
+echo '# output data' >>$LIBCONFIG_FILE
+echo 'o=`eval "echo $o"`' >>$LIBCONFIG_FILE
+echo 'o=`eval "echo $o"`' >>$LIBCONFIG_FILE
+echo 'if test -n "$o" ; then ' >>$LIBCONFIG_FILE
+echo '  eval "echo $o"' >>$LIBCONFIG_FILE
+echo 'fi' >>$LIBCONFIG_FILE
+echo ' ' >>$LIBCONFIG_FILE
+echo '# help text' >>$LIBCONFIG_FILE
+echo 'if test ! -z "$h" ; then ' >>$LIBCONFIG_FILE
+echo '  cat <<EOF' >>$LIBCONFIG_FILE
+echo 'All available options:' >>$LIBCONFIG_FILE
+echo '  --prefix=DIR and   change \$prefix and \$exec-prefix' >>$LIBCONFIG_FILE
+echo '  --exec-prefix=DIR  (affects all other options)' >>$LIBCONFIG_FILE
+echo '  --prefix           \$prefix        $prefix' >>$LIBCONFIG_FILE
+echo '  --exec_prefix  or... ' >>$LIBCONFIG_FILE
+echo '  --eprefix          \$exec_prefix   $exec_prefix' >>$LIBCONFIG_FILE
+echo '  --version          \$version       $version' >>$LIBCONFIG_FILE
+echo '  --cppflags         C preprocessor flags' >>$LIBCONFIG_FILE
+echo '  --cflags-only      C compiler flags' >>$LIBCONFIG_FILE
+echo '  --cflags           C preprocessor and compiler flags' >>$LIBCONFIG_FILE
+echo '  --libs             -L\$libdir \$LIBS $libs' >>$LIBCONFIG_FILE
+echo '  --package          \$package       $package' >>$LIBCONFIG_FILE
+echo '  --bindir           \$bindir        $bindir' >>$LIBCONFIG_FILE
+echo '  --sbindir          \$sbindir       $sbindir' >>$LIBCONFIG_FILE
+echo '  --libexecdir       \$libexecdir    $libexecdir' >>$LIBCONFIG_FILE
+echo '  --datadir          \$datadir       $datadir' >>$LIBCONFIG_FILE
+echo '  --sysconfdir       \$sysconfdir    $sysconfdir' >>$LIBCONFIG_FILE
+echo '  --sharedstatedir   \$sharedstatedir$sharedstatedir' >>$LIBCONFIG_FILE
+echo '  --localstatedir    \$localstatedir $localstatedir' >>$LIBCONFIG_FILE
+echo '  --libdir           \$libdir        $libdir' >>$LIBCONFIG_FILE
+echo '  --infodir          \$infodir       $infodir' >>$LIBCONFIG_FILE
+echo '  --mandir           \$mandir        $mandir' >>$LIBCONFIG_FILE
+echo '  --target           \$target        $target' >>$LIBCONFIG_FILE
+echo '  --host             \$host          $host' >>$LIBCONFIG_FILE
+echo '  --build            \$build         $build' >>$LIBCONFIG_FILE
+echo '  --pkgdatadir       \$datadir/\$package    ${datadir}/${package}'    >>$LIBCONFIG_FILE
+echo '  --pkglibdir        \$libdir/\$package     ${libdir}/${package}' >>$LIBCONFIG_FILE
+echo '  --pkgincludedir    \$includedir/\$package ${includedir}/${package}' >>$LIBCONFIG_FILE
+echo '  --template-version \$template_version     $template_version' >>$LIBCONFIG_FILE
+echo '  --help' >>$LIBCONFIG_FILE
+echo 'EOF' >>$LIBCONFIG_FILE
+echo 'fi' >>$LIBCONFIG_FILE
+m4_pushdef([LIBCONFIG_UP], [m4_translit([$1], [a-z], [A-Z])])dnl
+LIBCONFIG_UP[]_CONFIG="LIBCONFIG_DIR[]$1-config"
+AC_SUBST(LIBCONFIG_UP[]_CONFIG)
+dnl AC_CONFIG_FILES(LIBCONFIG_DIR[]$1[-config], [chmod +x ]LIBCONFIG_DIR[]$1[-config])
+m4_popdef([LIBCONFIG_DIR])
+m4_popdef([LIBCONFIG_UP])
+])
+
+
+# AC_CONFIG_LIBCONFIG_IN_MODULES [(MODULE1 [, MODULE2 [, ...]])]
+# --------------------------------------------------------------
+#Output shell script using custom module variables.
+AC_DEFUN([AC_CONFIG_LIBCONFIG_IN_MODULES],
+[m4_if([$1], , ,
+       [# create module definition for $1
+dnl we're going to need uppercase, lowercase and user-friendly versions of the
+dnl string `MODULE'
+m4_pushdef([MODULE_UP], m4_translit([$1], [a-z], [A-Z]))dnl
+m4_pushdef([MODULE_DOWN], m4_translit([$1], [A-Z], [a-z]))dnl
+if test -z "$MODULE_DOWN[]_cppflags" ; then
+  if test -n "$MODULE_UP[]_CPPFLAGS" ; then
+      MODULE_DOWN[]_cppflags="$MODULE_UP[]_CPPFLAGS"
+  else
+dnl    AC_MSG_WARN([variable `MODULE_DOWN[]_cppflags' undefined])
+    MODULE_DOWN[]_cppflags=''
+  fi
+fi
+AC_SUBST(MODULE_DOWN[]_cppflags)
+if test -z "$MODULE_DOWN[]_cflags" ; then
+  if test -n "$MODULE_UP[]_CFLAGS" ; then
+      MODULE_DOWN[]_cflags="$MODULE_UP[]_CFLAGS"
+  else
+dnl    AC_MSG_WARN([variable `MODULE_DOWN[]_cflags' undefined])
+    MODULE_DOWN[]_cflags=''
+  fi
+fi
+AC_SUBST(MODULE_DOWN[]_cflags)dnl
+if test -z "$MODULE_DOWN[]_libs" ; then
+  if test -n "$MODULE_UP[]_LIBS" ; then
+    MODULE_DOWN[]_libs="$MODULE_UP[]_LIBS"
+  else
+    AC_MSG_WARN([variable `MODULE_DOWN[]_libs' and `MODULE_UP[]_LIBS' undefined])
+    MODULE_DOWN[]_libs='-l$1'
+  fi
+  if test -n "$MODULE_UP[]_LIBDEPS" ; then
+    MODULE_DOWN[]_libs="$MODULE_DOWN[]_libs $MODULE_UP[]_LIBDEPS"
+  fi
+fi
+AC_SUBST(MODULE_DOWN[]_libs)dnl
+if test -z "$MODULE_UP[]_VERSION" ; then
+  AC_MSG_WARN([variable `MODULE_UP[]_VERSION' undefined])
+  MODULE_UP[]_VERSION="$VERSION"
+fi
+AC_SUBST(MODULE_UP[]_VERSION)dnl
+echo 'if test x$echo_module_$1 = xyes ; then' >>$LIBCONFIG_FILE
+AC_CONFIG_LIBCONFIG_IN_MODULES_VARS([cppflags], [MODULE_DOWN[]_cppflags], [cppflags])
+AC_CONFIG_LIBCONFIG_IN_MODULES_VARS([cflags], [MODULE_DOWN[]_cflags], [cflags])
+AC_CONFIG_LIBCONFIG_IN_MODULES_VARS([libs], [MODULE_DOWN[]_libs], [libs])
+AC_CONFIG_LIBCONFIG_IN_MODULES_VARS([version], [MODULE_UP[]_VERSION], [modversion])
+echo 'fi' >>$LIBCONFIG_FILE
+m4_popdef([MODULE_UP])dnl
+m4_popdef([MODULE_DOWN])dnl
+AC_CONFIG_LIBCONFIG_IN_MODULES(m4_shift($@))])dnl
+])
+
+
+# AC_CONFIG_LIBCONFIG_IN_MODULES_VARS [(VAR, SUBSTITUTION,
+# PKGCONFIG_ARGS)]
+# --------------------------------------------------------
+# Output AC_CONFIG_LIBCONFIG_IN_MODULES variables.
+#   VAR = variable to set
+#   SUBSTITUTION = set if pkg-config is not available
+#   PKGCONFIG_ARGS = args for pkg-config
+AC_DEFUN([AC_CONFIG_LIBCONFIG_IN_MODULES_VARS],
+[m4_if(AC_CONFIG_LIBCONFIG_IN_USEPKGCONFIG, [true],
+[
+echo 'if test -x "`which pkg-config`" ; then' >>$LIBCONFIG_FILE
+echo '  if pkg-config --atleast-pkgconfig-version=0.7 --exists "MODULE_DOWN"; then' >>$LIBCONFIG_FILE
+echo '    $1="@S|@$1 `pkg-config --$3 MODULE_DOWN`"' >>$LIBCONFIG_FILE
+echo '  fi' >>$LIBCONFIG_FILE
+echo 'else' >>$LIBCONFIG_FILE
+echo '  $1="@S|@$1 @$2@"' >>$LIBCONFIG_FILE
+echo 'fi' >>$LIBCONFIG_FILE
+],
+[echo '  $1="@S|@$1 @$2@"' >>$LIBCONFIG_FILE
+])])
+
+
+# AC_CONFIG_LIBCONFIG_IN_PKGCONFIG
+# --------------------------------
+# Enable pkgconfig support in libconfig script (default)
+AC_DEFUN([AC_CONFIG_LIBCONFIG_IN_PKGCONFIG],
+[m4_define([AC_CONFIG_LIBCONFIG_IN_USEPKGCONFIG], [true])
+])dnl
+
+
+# AC_CONFIG_LIBCONFIG_IN_STATIC
+# -----------------------------
+# Disable pkgconfig support in libconfig script
+AC_DEFUN([AC_CONFIG_LIBCONFIG_IN_STATIC],
+[m4_define([AC_CONFIG_LIBCONFIG_IN_USEPKGCONFIG], [false])
+])dnl
diff -ruN -xMakefile.in icecast2-2.3.1.orig/m4/ogg2.m4 icecast/m4/ogg2.m4
--- icecast2-2.3.1.orig/m4/ogg2.m4	1970-01-01 01:00:00.000000000 +0100
+++ icecast/m4/ogg2.m4	2006-06-20 17:28:32.000000000 +0200
@@ -0,0 +1,106 @@
+# Configure paths for libogg2
+# Arc Riley <arc@xiph.org> 03-03-2004
+# Shamelessly stolen from Jack Moffitt
+# Who shamelessly stole it from Owen Taylor and Manish Singh
+# Isn't Free Software awesome?
+
+dnl XIPH_PATH_OGG2([ACTION-IF-FOUND [, ACTION-IF-NOT-FOUND]])
+dnl Test for libogg, and define OGG_CFLAGS and OGG_LIBS
+dnl
+dnl $Id: ogg2.m4 7032 2004-07-07 06:02:32Z brendan $
+dnl
+AC_DEFUN([XIPH_PATH_OGG2],
+[dnl 
+dnl Get the cflags and libraries
+dnl
+AC_ARG_WITH(ogg,[  --with-ogg=PFX   Prefix where libogg2 is installed (optional)], ogg_prefix="$withval", ogg_prefix="")
+AC_ARG_WITH(ogg-libraries,[  --with-ogg-libraries=DIR   Directory where libogg2 library is installed (optional)], ogg_libraries="$withval", ogg_libraries="")
+AC_ARG_WITH(ogg-includes,[  --with-ogg-includes=DIR   Directory where libogg2 header files are installed (optional)], ogg_includes="$withval", ogg_includes="")
+AC_ARG_ENABLE(oggtest, [  --disable-oggtest       Do not try to compile and run a test Ogg program],, enable_oggtest=yes)
+
+  if test "x$ogg_libraries" != "x" ; then
+    OGG_LIBS="-L$ogg_libraries"
+  elif test "x$ogg_prefix" != "x" ; then
+    OGG_LIBS="-L$ogg_prefix/lib"
+  elif test "x$prefix" != "xNONE" ; then
+    OGG_LIBS="-L$prefix/lib"
+  fi
+
+  OGG_LIBS="$OGG_LIBS -logg2"
+
+  if test "x$ogg_includes" != "x" ; then
+    OGG_CFLAGS="-I$ogg_includes"
+  elif test "x$ogg_prefix" != "x" ; then
+    OGG_CFLAGS="-I$ogg_prefix/include"
+  elif test "$prefix" != "xNONE"; then
+    OGG_CFLAGS="-I$prefix/include"
+  fi
+
+  AC_MSG_CHECKING(for Ogg2)
+  no_ogg=""
+
+
+  if test "x$enable_oggtest" = "xyes" ; then
+    ac_save_CFLAGS="$CFLAGS"
+    ac_save_LIBS="$LIBS"
+    CFLAGS="$CFLAGS $OGG_CFLAGS"
+    LIBS="$LIBS $OGG_LIBS"
+dnl
+dnl Now check if the installed Ogg is sufficiently new.
+dnl
+      rm -f conf.oggtest
+      AC_TRY_RUN([
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <ogg2/ogg.h>
+
+int main ()
+{
+  system("touch conf.oggtest");
+  return 0;
+}
+
+],, no_ogg=yes,[echo $ac_n "cross compiling; assumed OK... $ac_c"])
+       CFLAGS="$ac_save_CFLAGS"
+       LIBS="$ac_save_LIBS"
+  fi
+
+  if test "x$no_ogg" = "x" ; then
+     AC_MSG_RESULT(yes)
+     ifelse([$1], , :, [$1])     
+  else
+     AC_MSG_RESULT(no)
+     if test -f conf.oggtest ; then
+       :
+     else
+       echo "*** Could not run Ogg test program, checking why..."
+       CFLAGS="$CFLAGS $OGG_CFLAGS"
+       LIBS="$LIBS $OGG_LIBS"
+       AC_TRY_LINK([
+#include <stdio.h>
+#include <ogg2/ogg.h>
+],     [ return 0; ],
+       [ echo "*** The test program compiled, but did not run. This usually means"
+       echo "*** that the run-time linker is not finding Ogg or finding the wrong"
+       echo "*** version of Ogg. If it is not finding Ogg, you'll need to set your"
+       echo "*** LD_LIBRARY_PATH environment variable, or edit /etc/ld.so.conf to point"
+       echo "*** to the installed location  Also, make sure you have run ldconfig if that"
+       echo "*** is required on your system"
+       echo "***"
+       echo "*** If you have an old version installed, it is best to remove it, although"
+       echo "*** you may also be able to get things to work by modifying LD_LIBRARY_PATH"],
+       [ echo "*** The test program failed to compile or link. See the file config.log for the"
+       echo "*** exact error that occured. This usually means Ogg was incorrectly installed"
+       echo "*** or that you have moved Ogg since it was installed." ])
+       CFLAGS="$ac_save_CFLAGS"
+       LIBS="$ac_save_LIBS"
+     fi
+     OGG_CFLAGS=""
+     OGG_LIBS=""
+     ifelse([$2], , :, [$2])
+  fi
+  AC_SUBST(OGG_CFLAGS)
+  AC_SUBST(OGG_LIBS)
+  rm -f conf.oggtest
+])
diff -ruN -xMakefile.in icecast2-2.3.1.orig/m4/shout.m4 icecast/m4/shout.m4
--- icecast2-2.3.1.orig/m4/shout.m4	1970-01-01 01:00:00.000000000 +0100
+++ icecast/m4/shout.m4	2006-06-20 17:28:32.000000000 +0200
@@ -0,0 +1,79 @@
+dnl XIPH_PATH_SHOUT
+dnl Jack Moffitt <jack@icecast.org> 08-06-2001
+dnl Rewritten for libshout 2
+dnl Brendan Cully <brendan@xiph.org> 20030612
+dnl 
+dnl $Id: shout.m4 7180 2004-07-20 02:50:54Z brendan $
+
+# XIPH_PATH_SHOUT([ACTION-IF-FOUND [, ACTION-IF-NOT-FOUND]])
+# Test for libshout, and define SHOUT_CPPFLAGS SHOUT_CFLAGS SHOUT_LIBS, and
+# SHOUT_THREADSAFE
+AC_DEFUN([XIPH_PATH_SHOUT],
+[dnl
+xt_have_shout="no"
+SHOUT_THREADSAFE="no"
+SHOUT_CPPFLAGS=""
+SHOUT_CFLAGS=""
+SHOUT_LIBS=""
+
+# NB: PKG_CHECK_MODULES exits if pkg-config is unavailable on the target
+# system, so we can't use it.
+
+# seed pkg-config with the default libshout location
+PKG_CONFIG_PATH=${PKG_CONFIG_PATH:-/usr/local/lib/pkgconfig}
+export PKG_CONFIG_PATH
+
+# Step 1: Use pkg-config if available
+AC_PATH_PROG([PKGCONFIG], [pkg-config], [no])
+if test "$PKGCONFIG" != "no" && `$PKGCONFIG --exists shout`
+then
+  SHOUT_CFLAGS=`$PKGCONFIG --variable=cflags_only shout`
+  SHOUT_CPPFLAGS=`$PKGCONFIG --variable=cppflags shout`
+  SHOUT_LIBS=`$PKGCONFIG --libs shout`
+  xt_have_shout="maybe"
+else
+  if test "$PKGCONFIG" != "no"
+  then
+    AC_MSG_NOTICE([$PKGCONFIG couldn't find libshout. Try adjusting PKG_CONFIG_PATH.])
+  fi
+  # pkg-config unavailable, try shout-config
+  AC_PATH_PROG([SHOUTCONFIG], [shout-config], [no])
+  if test "$SHOUTCONFIG" != "no" && test `$SHOUTCONFIG --package` = "libshout"
+  then
+    SHOUT_CPPFLAGS=`$SHOUTCONFIG --cppflags`
+    SHOUT_CFLAGS=`$SHOUTCONFIG --cflags-only`
+    SHOUT_LIBS=`$SHOUTCONFIG --libs`
+    xt_have_shout="maybe"
+  fi
+fi
+
+# Now try actually using libshout
+if test "$xt_have_shout" != "no"
+then
+  ac_save_CPPFLAGS="$CPPFLAGS"
+  ac_save_CFLAGS="$CFLAGS"
+  ac_save_LIBS="$LIBS"
+  CPPFLAGS="$CPPFLAGS $SHOUT_CPPFLAGS"
+  CFLAGS="$CFLAGS $SHOUT_CFLAGS"
+  LIBS="$SHOUT_LIBS $LIBS"
+  AC_CHECK_HEADERS([shout/shout.h], [
+    AC_CHECK_FUNC([shout_new], [
+      ifelse([$1], , :, [$1])
+      xt_have_shout="yes"
+    ])
+    AC_EGREP_CPP([yes], [#include <shout/shout.h>
+#if SHOUT_THREADSAFE
+yes
+#endif
+], [SHOUT_THREADSAFE="yes"])
+  ])
+  CPPFLAGS="$ac_save_CPPFLAGS"
+  CFLAGS="$ac_save_CFLAGS"
+  LIBS="$ac_save_LIBS"
+fi
+
+if test "$xt_have_shout" != "yes"
+then
+  ifelse([$2], , :, [$2])
+fi
+])dnl XIPH_PATH_SHOUT
diff -ruN -xMakefile.in icecast2-2.3.1.orig/m4/xiph_openssl.m4 icecast/m4/xiph_openssl.m4
--- icecast2-2.3.1.orig/m4/xiph_openssl.m4	1970-01-01 01:00:00.000000000 +0100
+++ icecast/m4/xiph_openssl.m4	2006-06-20 17:28:32.000000000 +0200
@@ -0,0 +1,49 @@
+dnl XIPH_PATH_OPENSSL([ACTION-IF-FOUND [, ACTION-IF-NOT-FOUND]])
+dnl Karl Heyes <karl@xiph.org>
+dnl Test for openssl, and define OPENSSL_CFLAGS and OPENSSL_LIBS
+dnl
+AC_DEFUN([XIPH_PATH_OPENSSL],
+[dnl 
+dnl Get the cflags and libraries
+dnl
+AC_ARG_WITH(openssl,
+    AC_HELP_STRING([--with-openssl=PFX],[Prefix where openssl is installed (optional)]),
+    openssl_prefix="$withval", openssl_prefix="")
+
+if test "x$openssl_prefix" != "x" -a "x$openssl_prefix" != "xyes"; then
+    OPENSSL_LIBS="-L$openssl_prefix/lib -lssl"
+    OPENSSL_CFLAGS="-I$openssl_prefix/include"
+else
+    AC_PATH_PROG([PKGCONFIG], [pkg-config], [no])
+    if test "$PKGCONFIG" != "no" && `$PKGCONFIG --exists openssl`; then
+        OPENSSL_CFLAGS=`$PKGCONFIG --cflags openssl`
+        OPENSSL_LIBS=`$PKGCONFIG --libs openssl`
+    else
+        if test "x$prefix" = "xNONE"; then
+            openssl_prefix="/usr/local"
+        else
+            openssl_prefix="$prefix"
+        fi
+        OPENSSL_LIBS="-L$openssl_prefix/lib -lssl"
+        OPENSSL_CFLAGS="-I$openssl_prefix/include"
+    fi
+fi
+
+# Now try linking to openssl
+xt_save_CFLAGS="$CFLAGS"
+xt_save_LIBS="$LIBS"
+CFLAGS="$CFLAGS $OPENSSL_CFLAGS"
+LIBS="$OPENSSL_LIBS $LIBS"
+AC_TRY_LINK([#include <openssl/ssl.h>], [void *a = SSL_new], [openssl_ok='yes'])
+CFLAGS="$xt_save_CFLAGS"
+LIBS="$xt_save_LIBS"
+
+if test "$openssl_ok" = "yes"; then
+    AC_DEFINE(HAVE_OPENSSL, 1, [Define if you have libopenssl.])
+    ifelse([$1], , :, [$1])     
+else
+    OPENSSL_LIBS=""
+    OPENSSL_CFLAGS=""
+    ifelse([$2], , :, [$2])
+fi
+])
diff -ruN -xMakefile.in icecast2-2.3.1.orig/m4/xiph_path_python.m4 icecast/m4/xiph_path_python.m4
--- icecast2-2.3.1.orig/m4/xiph_path_python.m4	1970-01-01 01:00:00.000000000 +0100
+++ icecast/m4/xiph_path_python.m4	2006-06-20 17:28:32.000000000 +0200
@@ -0,0 +1,73 @@
+dnl local M4 configure macros
+dnl Brendan Cully <brendan@xiph.org>
+dnl $Id: xiph_path_python.m4 7020 2004-07-06 19:21:46Z brendan $
+
+# XIPH_PATH_PYTHON(PATH)
+# Search for python in PATH, or in the path if none is given.
+# Defines PYTHON_CPPFLAGS and PYTHON_LIBS if found
+AC_DEFUN([XIPH_PATH_PYTHON],
+  [
+m4_pushdef([xpp_path], [$1])
+
+PYTHON="no"
+
+if test "xpp_path" != "yes"
+then
+  AC_MSG_CHECKING([python])
+  if test -x "xpp_path"
+  then
+    PYTHON="xpp_path"
+  fi
+  AC_MSG_RESULT([$PYTHON])
+else
+  AC_PATH_PROGS([PYTHON], [python python2 python2.3 python2.2])
+fi
+
+m4_popdef([xpp_path])
+
+if test "$PYTHON" != "no"
+then
+  # The library we're linking against
+  PYTHON_LIB="_XIPH_PYTHON_CFG([$PYTHON], [LIBRARY])"
+  
+  # if LIBRARY is nonsensical, bail out
+  if test $? -ne 0 -o -z "$PYTHON_LIB"
+  then
+    AC_MSG_WARN([Could not find library for $PYTHON])
+    break
+  fi
+  # make library linker friendly. This is a hack, but I don't know what's better
+  PYTHON_LIB=`echo "$PYTHON_LIB" | sed 's/lib//;s/\.a.*//;s/\.so.*//;s/\.dylib.*//'`
+
+  # LDFLAGS
+  PYTHON_LDFLAGS="-L[]_XIPH_PYTHON_CFG([$PYTHON], [LIBPL])"
+
+  # Extra libraries required by python
+  PYTHON_EXTRA_LIBS="_XIPH_PYTHON_CFG([$PYTHON], [LIBS])"
+  PYTHON_EXTRA_LIBS="$PYTHON_EXTRA_LIBS _XIPH_PYTHON_CFG([$PYTHON], [SYSLIBS])"
+  PYTHON_EXTRA_LIBS="$PYTHON_EXTRA_LIBS _XIPH_PYTHON_CFG([$PYTHON], [SHLIBS])"
+
+  PYTHON_CPPFLAGS="-I[]_XIPH_PYTHON_CFG([$PYTHON], [INCLUDEPY])"
+
+  # test header and library functionality
+  saved_CPPFLAGS="$CPPFLAGS"
+  CPPFLAGS="$CPPFLAGS $PYTHON_CPPFLAGS"
+  AC_CHECK_HEADER([Python.h],
+    [
+    saved_LDFLAGS="$LDFLAGS"
+    LDFLAGS="$LDFLAGS $PYTHON_LDFLAGS"
+    AC_CHECK_LIB([$PYTHON_LIB], [Py_Initialize],
+      [PYTHON_LIBS="-l$PYTHON_LIB $PYTHON_EXTRA_LIBS"],
+      [AC_MSG_WARN([Could not link to the python library])],
+      [$PYTHON_EXTRA_LIBS])
+    LDFLAGS="$saved_LDFLAGS"
+    ],
+    [AC_MSG_WARN([Python.h doesn't appear to be usable])])
+  CPPFLAGS="$saved_CPPFLAGS"
+fi
+  ])dnl
+
+# _XIPH_PYTHON_CFG(PYTHONPATH, CFGVAR)
+# Ask python in PYTHONPATH for the definition of CFGVAR
+m4_define([_XIPH_PYTHON_CFG],
+  [`$1 -c 'from distutils.sysconfig import get_config_var; print get_config_var("$2")' | sed 's/None//'`])
diff -ruN -xMakefile.in icecast2-2.3.1.orig/README icecast/README
--- icecast2-2.3.1.orig/README	2005-08-19 04:02:00.000000000 +0200
+++ icecast/README	2006-06-20 17:28:20.000000000 +0200
@@ -5,7 +5,7 @@
 Vorbis and MP3 audio streams. It can be used to create an Internet 
 radio station or a privately running jukebox and many things in 
 between. It is very versatile in that new formats can be added 
-relatively easily and supports open standards for commuincation and 
+relatively easily and supports open standards for communication and 
 interaction.
 
 Icecast is distributed under the GNU GPL, version 2. A copy of this
diff -ruN -xMakefile.in icecast2-2.3.1.orig/src/admin.c icecast/src/admin.c
--- icecast2-2.3.1.orig/src/admin.c	2005-09-13 00:50:48.000000000 +0200
+++ icecast/src/admin.c	2006-06-20 17:28:14.000000000 +0200
@@ -30,7 +30,7 @@
 #include "global.h"
 #include "event.h"
 #include "stats.h"
-#include "os.h"
+#include "compat.h"
 #include "xslt.h"
 #include "fserve.h"
 
@@ -267,7 +267,7 @@
     return(doc);
 }
 
-void admin_send_response(xmlDocPtr doc, client_t *client, 
+static void admin_send_response(xmlDocPtr doc, client_t *client, 
         int response, char *xslt_template)
 {
     if (response == RAW)
diff -ruN -xMakefile.in icecast2-2.3.1.orig/src/auth.c icecast/src/auth.c
--- icecast2-2.3.1.orig/src/auth.c	2005-11-30 19:16:17.000000000 +0100
+++ icecast/src/auth.c	2006-06-20 17:28:14.000000000 +0200
@@ -286,6 +286,12 @@
         if (loop && source->fallback_when_full && source->fallback_mount)
         {
             source_t *next = source_find_mount (source->fallback_mount);
+            if (!next) {
+                ERROR2("Fallback '%s' for full source '%s' not found", 
+                        source->mount, source->fallback_mount);
+                return -1;
+            }
+
             INFO1 ("stream full trying %s", next->mount);
             source = next;
             loop--;
@@ -312,7 +318,6 @@
         /* enable on-demand relay to start, wake up the slave thread */
         DEBUG0("kicking off on-demand relay");
         source->on_demand_req = 1;
-        slave_rescan ();
     }
     DEBUG1 ("Added client to %s", source->mount);
     return 0;
@@ -580,7 +585,7 @@
 
 /* these are called at server start and termination */
 
-void auth_initialise ()
+void auth_initialise (void)
 {
     clients_to_auth = NULL;
     auth_pending_count = 0;
@@ -589,7 +594,7 @@
     auth_thread = thread_create ("auth thread", auth_run_thread, NULL, THREAD_ATTACHED);
 }
 
-void auth_shutdown ()
+void auth_shutdown (void)
 {
     if (auth_thread)
     {
diff -ruN -xMakefile.in icecast2-2.3.1.orig/src/auth.h icecast/src/auth.h
--- icecast2-2.3.1.orig/src/auth.h	2005-09-08 16:03:51.000000000 +0200
+++ icecast/src/auth.h	2006-06-20 17:28:14.000000000 +0200
@@ -35,7 +35,7 @@
     AUTH_FORBIDDEN,
     AUTH_USERADDED,
     AUTH_USEREXISTS,
-    AUTH_USERDELETED,
+    AUTH_USERDELETED
 } auth_result;
 
 typedef struct auth_client_tag
@@ -77,8 +77,8 @@
 void add_client (const char *mount, client_t *client);
 int  release_client (client_t *client);
 
-void auth_initialise ();
-void auth_shutdown ();
+void auth_initialise (void);
+void auth_shutdown (void);
 
 auth_t  *auth_get_authenticator (xmlNodePtr node);
 void    auth_release (auth_t *authenticator);
diff -ruN -xMakefile.in icecast2-2.3.1.orig/src/avl/avl.c icecast/src/avl/avl.c
--- icecast2-2.3.1.orig/src/avl/avl.c	2005-08-19 04:01:58.000000000 +0200
+++ icecast/src/avl/avl.c	2006-06-20 17:28:41.000000000 +0200
@@ -70,6 +70,7 @@
   } else {
     avl_node * root = avl_node_new((void *)NULL, (avl_node *) NULL);
     if (!root) {
+      free (t);
       return NULL;
     } else {
       t->root = root;
diff -ruN -xMakefile.in icecast2-2.3.1.orig/src/cfgfile.c icecast/src/cfgfile.c
--- icecast2-2.3.1.orig/src/cfgfile.c	2005-10-24 15:48:20.000000000 +0200
+++ icecast/src/cfgfile.c	2006-06-20 17:28:14.000000000 +0200
@@ -86,12 +86,12 @@
         ice_config_t *c);
 static void _add_server(xmlDocPtr doc, xmlNodePtr node, ice_config_t *c);
 
-static void create_locks() {
+static void create_locks(void) {
     thread_mutex_create(&_locks.relay_lock);
     thread_rwlock_create(&_locks.config_lock);
 }
 
-static void release_locks() {
+static void release_locks(void) {
     thread_mutex_destroy(&_locks.relay_lock);
     thread_rwlock_destroy(&_locks.config_lock);
 }
@@ -1039,6 +1039,9 @@
         server = NULL;
         addnode = 0;
     }
+    else {
+        free (server);
+    }
     
 }
 
diff -ruN -xMakefile.in icecast2-2.3.1.orig/src/compat.h icecast/src/compat.h
--- icecast2-2.3.1.orig/src/compat.h	2005-08-19 04:01:58.000000000 +0200
+++ icecast/src/compat.h	2006-06-20 17:28:14.000000000 +0200
@@ -20,12 +20,23 @@
  * Solaris.
  */
 
+#ifdef _WIN32
+#include <windows.h>
+#else
+#include <unistd.h>
+#endif
+
+
 /* Make sure we define 64 bit types */
 #ifdef _WIN32
+#  define PATH_SEPARATOR "\\"
+#  define size_t unsigned int
+#  define ssize_t int
 #  define int64_t __int64
 #  define uint64_t unsigned __int64
 #  define uint32_t unsigned int
 #else
+#  define PATH_SEPARATOR "/"
 #  if defined(HAVE_STDINT_H)
 #    include <stdint.h>
 #  elif defined(HAVE_INTTYPES_H)
diff -ruN -xMakefile.in icecast2-2.3.1.orig/src/configtest.c icecast/src/configtest.c
--- icecast2-2.3.1.orig/src/configtest.c	1970-01-01 01:00:00.000000000 +0100
+++ icecast/src/configtest.c	2006-06-20 17:28:14.000000000 +0200
@@ -0,0 +1,75 @@
+/* Icecast
+ *
+ * This program is distributed under the GNU General Public License, version 2.
+ * A copy of this license is included with this source.
+ *
+ * Copyright 2000-2004, Jack Moffitt <jack@xiph.org, 
+ *                      Michael Smith <msmith@xiph.org>,
+ *                      oddsock <oddsock@xiph.org>,
+ *                      Karl Heyes <karl@xiph.org>
+ *                      and others (see AUTHORS for details).
+ */
+
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#include <stdio.h>
+#include "cfgfile.h"
+
+void _dump_config(ice_config_t *config);
+
+int main(void)
+{
+    ice_config_t *config;
+
+    config_initialize();
+    
+    config_parse_file("icecast.xml");
+
+    config = config_get_config_unlocked();
+
+    _dump_config(config);
+
+    config_shutdown();
+
+    return 0;
+}
+
+void _dump_config(ice_config_t *config)
+{
+    ice_config_dir_t *node;
+
+    printf("-----\n");
+    printf("location = %s\n", config->location);
+    printf("admin = %s\n", config->admin);
+    printf("client_limit = %d\n", config->client_limit);
+    printf("source_limit = %d\n", config->source_limit);
+    printf("threadpool_size = %d\n", config->threadpool_size);
+    printf("client_timeout = %d\n", config->client_timeout);
+    printf("source_password = %s\n", config->source_password);
+    printf("touch_interval = %d\n", config->touch_interval);
+
+    node = config->dir_list;
+    while (node) {
+        printf("directory.touch_interval = %d\n", node->touch_interval);
+        printf("directory.host = %s\n", node->host);
+        
+        node = node->next;
+    }
+
+    printf("hostname = %s\n", config->hostname);
+    printf("port = %d\n", config->port);
+    printf("bind_address = %s\n", config->bind_address);
+    printf("base_dir = %s\n", config->base_dir);
+    printf("log_dir = %s\n", config->log_dir);
+    printf("access_log = %s\n", config->access_log);
+    printf("error_log = %s\n", config->error_log);
+    printf("loglevel = %d\n", config->loglevel);
+    printf("-----\n");
+}
+
+
+
+
+
diff -ruN -xMakefile.in icecast2-2.3.1.orig/src/connection.c icecast/src/connection.c
--- icecast2-2.3.1.orig/src/connection.c	2005-11-30 19:16:17.000000000 +0100
+++ icecast/src/connection.c	2006-06-20 17:28:14.000000000 +0200
@@ -34,7 +34,7 @@
 #define strncasecmp strnicmp
 #endif
 
-#include "os.h"
+#include "compat.h"
 
 #include "thread/thread.h"
 #include "avl/avl.h"
@@ -117,6 +117,10 @@
     thread_mutex_create(&move_clients_mutex);
     thread_rwlock_create(&_source_shutdown_rwlock);
     thread_cond_create(&global.shutdown_cond);
+    _req_queue = NULL;
+    _req_queue_tail = &_req_queue;
+    _con_queue = NULL;
+    _con_queue_tail = &_con_queue;
 
     _initialized = 1;
 }
@@ -313,7 +317,7 @@
 
 
 /* run along queue checking for any data that has come in or a timeout */
-static void process_request_queue ()
+static void process_request_queue (void)
 {
     client_queue_t **node_ref = (client_queue_t **)&_req_queue;
     ice_config_t *config = config_get_config ();
@@ -386,6 +390,7 @@
                 *node_ref = node->next;
                 node->next = NULL;
                 _add_connection (node);
+                continue;
             }
         }
         else
@@ -962,8 +967,10 @@
         client->parser = parser;
         _handle_source_request (client, shoutcast_mount, SHOUTCAST_SOURCE_AUTH);
     }
-    else
+    else {
+        httpp_destroy (parser);
         client_destroy (client);
+    }
     free (http_compliant);
     free (shoutcast_mount);
     free (node);
diff -ruN -xMakefile.in icecast2-2.3.1.orig/src/format.c icecast/src/format.c
--- icecast2-2.3.1.orig/src/format.c	2005-09-08 16:03:51.000000000 +0200
+++ icecast/src/format.c	2006-06-20 17:28:14.000000000 +0200
@@ -93,8 +93,11 @@
 {
     refbuf_t *refbuf = source->burst_point;
 
-    /* we only want to attempt a burst at connection time, not midstream */
-    if (client->intro_offset == -1)
+    /* we only want to attempt a burst at connection time, not midstream 
+     * however streams like theora may not have the most recent page marked as
+     * a starting point, so look for one from the burst point */
+    if (client->intro_offset == -1 && source->stream_data_tail
+            && source->stream_data_tail->sync_point)
         refbuf = source->stream_data_tail;
     else
     {
diff -ruN -xMakefile.in icecast2-2.3.1.orig/src/format_flac.c icecast/src/format_flac.c
--- icecast2-2.3.1.orig/src/format_flac.c	2005-08-19 04:01:58.000000000 +0200
+++ icecast/src/format_flac.c	2006-06-20 17:28:14.000000000 +0200
@@ -46,14 +46,14 @@
 {
     refbuf_t * refbuf;
 
-    if (ogg_stream_pagein (&codec->os, page) < 0)
-    {
-        ogg_info->error = 1;
-        return NULL;
-    }
     if (codec->headers)
     {
         ogg_packet packet;
+        if (ogg_stream_pagein (&codec->os, page) < 0)
+        {
+            ogg_info->error = 1;
+            return NULL;
+        }
         while (ogg_stream_packetout (&codec->os, &packet))
         {
             int type = packet.packet[0];
diff -ruN -xMakefile.in icecast2-2.3.1.orig/src/format_ogg.c icecast/src/format_ogg.c
--- icecast2-2.3.1.orig/src/format_ogg.c	2005-08-19 04:01:58.000000000 +0200
+++ icecast/src/format_ogg.c	2006-06-20 17:28:14.000000000 +0200
@@ -181,7 +181,7 @@
 }
 
 
-void format_ogg_free_plugin (format_plugin_t *plugin)
+static void format_ogg_free_plugin (format_plugin_t *plugin)
 {
     ogg_state_t *state = plugin->_state;
 
@@ -240,6 +240,7 @@
 
         /* any others */
         ERROR0 ("Seen BOS page with unknown type");
+        ogg_info->error = 1;
         return -1;
     } while (0);
 
diff -ruN -xMakefile.in icecast2-2.3.1.orig/src/format_vorbis.c icecast/src/format_vorbis.c
--- icecast2-2.3.1.orig/src/format_vorbis.c	2005-11-30 19:16:17.000000000 +0100
+++ icecast/src/format_vorbis.c	2006-06-20 17:28:14.000000000 +0200
@@ -331,7 +331,7 @@
             vorbis_comment_add_tag (&vc, "artist", ogg_info->artist);
         if (ogg_info->title)
             vorbis_comment_add_tag (&vc, "title", ogg_info->title);
-        vorbis_comment_add (&vc, "server=" ICECAST_VERSION_STRING);
+        vorbis_comment_add_tag (&vc, "server", ICECAST_VERSION_STRING);
         vorbis_commentheader_out (&vc, &header);
 
         ogg_stream_packetin (&source_vorbis->new_os, &header);
diff -ruN -xMakefile.in icecast2-2.3.1.orig/src/fserve.c icecast/src/fserve.c
--- icecast2-2.3.1.orig/src/fserve.c	2005-09-08 16:03:51.000000000 +0200
+++ icecast/src/fserve.c	2006-06-20 17:28:14.000000000 +0200
@@ -201,7 +201,7 @@
 }
 #endif
 
-static void wait_for_fds() {
+static void wait_for_fds(void) {
     fserve_t *fclient;
 
     while (run_fserv)
@@ -465,13 +465,14 @@
     }
 
     file = fopen (fullpath, "rb");
-    free (fullpath);
     if (file == NULL)
     {
         WARN1 ("Problem accessing file \"%s\"", fullpath);
         client_send_404 (httpclient, "File not readable");
+        free (fullpath);
         return -1;
     }
+    free (fullpath);
 
     content_length = (int64_t)file_buf.st_size;
     range = httpp_getvar (httpclient->parser, "range");
@@ -526,21 +527,11 @@
                     fserve_content_type(path));
             }
             else {
-                httpclient->respcode = 416;
-                sock_write (httpclient->con->sock,
-                    "HTTP/1.0 416 Request Range Not Satisfiable\r\n\r\n");
-                client_destroy (httpclient);
-                return -1;
+                goto fail;
             }
         }
         else {
-            /* If we run into any issues with the ranges
-               we fallback to a normal/non-range request */
-            httpclient->respcode = 416;
-            sock_write (httpclient->con->sock,
-                "HTTP/1.0 416 Request Range Not Satisfiable\r\n\r\n");
-            client_destroy (httpclient);
-            return -1;
+            goto fail;
         }
     }
     else {
@@ -560,6 +551,14 @@
     fserve_add_client (httpclient, file);
 
     return 0;
+
+fail:
+    fclose (file);
+    httpclient->respcode = 416;
+    sock_write (httpclient->con->sock, 
+            "HTTP/1.0 416 Request Range Not Satisfiable\r\n\r\n");
+    client_destroy (httpclient);
+    return -1;
 }
 
 
diff -ruN -xMakefile.in icecast2-2.3.1.orig/src/httpp/BUILDING icecast/src/httpp/BUILDING
--- icecast2-2.3.1.orig/src/httpp/BUILDING	1970-01-01 01:00:00.000000000 +0100
+++ icecast/src/httpp/BUILDING	2006-06-20 17:28:51.000000000 +0200
@@ -0,0 +1,7 @@
+defines that affect compilation
+
+none
+
+library dependencies
+
+uses avl
diff -ruN -xMakefile.in icecast2-2.3.1.orig/src/httpp/test.c icecast/src/httpp/test.c
--- icecast2-2.3.1.orig/src/httpp/test.c	1970-01-01 01:00:00.000000000 +0100
+++ icecast/src/httpp/test.c	2006-06-20 17:28:51.000000000 +0200
@@ -0,0 +1,59 @@
+#include <stdio.h>
+
+#include <avl/avl.h>
+#include "httpp.h"
+
+
+int main(int argc, char **argv)
+{
+    char buff[8192];
+    int readed;
+    http_parser_t parser;
+    avl_node *node;
+    http_var_t *var;
+
+    httpp_initialize(&parser, NULL);
+
+    readed = fread(buff, 1, 8192, stdin);
+    if (httpp_parse(&parser, buff, readed)) {
+        printf("Parse succeeded...\n\n");
+        printf("Request was ");
+        switch (parser.req_type) {
+        case httpp_req_none:
+            printf(" none\n");
+            break;
+        case httpp_req_unknown:
+            printf(" unknown\n");
+            break;
+        case httpp_req_get:
+            printf(" get\n");
+            break;
+        case httpp_req_post:
+            printf(" post\n");
+            break;
+        case httpp_req_head:
+            printf(" head\n");
+            break;
+        }
+        printf("Version was 1.%d\n", parser.version);
+        
+        node = avl_get_first(parser.vars);
+        while (node) {
+            var = (http_var_t *)node->key;
+            
+            if (var)
+                printf("Iterating variable(s): %s = %s\n", var->name, var->value);
+            
+            node = avl_get_next(node);
+        }
+    } else {
+        printf("Parse failed...\n");
+    }
+
+    printf("Destroying parser...\n");
+    httpp_destroy(&parser);
+
+    return 0;
+}
+
+
diff -ruN -xMakefile.in icecast2-2.3.1.orig/src/log/log.c icecast/src/log/log.c
--- icecast2-2.3.1.orig/src/log/log.c	2005-10-24 16:41:50.000000000 +0200
+++ icecast/src/log/log.c	2006-06-20 17:29:06.000000000 +0200
@@ -46,6 +46,14 @@
 static mutex_t _logger_mutex;
 static int _initialized = 0;
 
+typedef struct _log_entry_t
+{
+   char *line;
+   unsigned int len;
+   struct _log_entry_t *next;
+} log_entry_t;
+
+
 typedef struct log_tag
 {
     int in_use;
@@ -57,16 +65,22 @@
     off_t size;
     off_t trigger_level;
     int archive_timestamp;
+
+    unsigned long total;
+    unsigned int entries;
+    unsigned int keep_entries;
+    log_entry_t *log_head;
+    log_entry_t **log_tail;
     
     char *buffer;
 } log_t;
 
 static log_t loglist[LOG_MAXLOGS];
 
-static int _get_log_id();
+static int _get_log_id(void);
 static void _release_log_id(int log_id);
-static void _lock_logger();
-static void _unlock_logger();
+static void _lock_logger(void);
+static void _unlock_logger(void);
 
 
 static int _log_open (int id, const char *file_timestamp)
@@ -115,7 +129,7 @@
     return 1;
 }
 
-void log_initialize()
+void log_initialize(void)
 {
     int i;
 
@@ -129,6 +143,11 @@
         loglist[i].filename = NULL;
         loglist[i].logfile = NULL;
         loglist[i].buffer = NULL;
+        loglist[i].total = 0;
+        loglist[i].entries = 0;
+        loglist[i].keep_entries = 0;
+        loglist[i].log_head = NULL;
+        loglist[i].log_tail = &loglist[i].log_head;
     }
 
     /* initialize mutexes */
@@ -178,6 +197,9 @@
         loglist [id] . filename = strdup (filename);
         if (stat (loglist [id] . filename, &st) == 0)
             loglist [id] . size = st.st_size;
+        loglist [id] . entries = 0;
+        loglist [id] . log_head = NULL;
+        loglist [id] . log_tail = &loglist [id] . log_head;
     }
 
     return id;
@@ -198,7 +220,8 @@
 {
     if (id < 0 || id >= LOG_MAXLOGS)
         return LOG_EINSANE;
-    if (!strcmp(filename, "") || loglist [id] . in_use == 0)
+    /* NULL filename is ok, empty filename is not. */
+    if ((filename && !strcmp(filename, "")) || loglist [id] . in_use == 0)
         return LOG_EINSANE;
      _lock_logger();
     if (loglist [id] . filename)
@@ -229,6 +252,26 @@
 }
 
 
+void log_set_lines_kept (int log_id, unsigned int count)
+{
+    if (log_id < 0 || log_id >= LOG_MAXLOGS) return;
+    if (loglist[log_id].in_use == 0) return;
+
+    _lock_logger ();
+    loglist[log_id].keep_entries = count;
+    while (loglist[log_id].entries > count)
+    {
+        log_entry_t *to_go = loglist [log_id].log_head;
+        loglist [log_id].log_head = to_go->next;
+        loglist [log_id].total -= to_go->len;
+        free (to_go->line);
+        free (to_go);
+        loglist [log_id].entries--;
+    }
+    _unlock_logger ();
+}
+
+
 void log_set_level(int log_id, unsigned level)
 {
     if (log_id < 0 || log_id >= LOG_MAXLOGS) return;
@@ -285,10 +328,19 @@
         fclose (loglist [log_id] . logfile);
         loglist [log_id] . logfile = NULL;
     }
+    while (loglist[log_id].entries)
+    {
+        log_entry_t *to_go = loglist [log_id].log_head;
+        loglist [log_id].log_head = to_go->next;
+        loglist [log_id].total -= to_go->len;
+        free (to_go->line);
+        free (to_go);
+        loglist [log_id].entries--;
+    }
     _unlock_logger();
 }
 
-void log_shutdown()
+void log_shutdown(void)
 {
     /* destroy mutexes */
 #ifndef _WIN32
@@ -300,19 +352,81 @@
     _initialized = 0;
 }
 
+
+static int create_log_entry (int log_id, const char *pre, const char *line)
+{
+    int len;
+    log_entry_t *entry;
+
+    if (loglist[log_id].keep_entries == 0)
+        return fprintf (loglist[log_id].logfile, "%s%s\n", pre, line); 
+    
+    entry = calloc (1, sizeof (log_entry_t));
+    entry->line = malloc (LOG_MAXLINELEN);
+    len = snprintf (entry->line, LOG_MAXLINELEN, "%s%s\n", pre, line);
+    entry->len = len;
+    loglist [log_id].total += len;
+    fprintf (loglist[log_id].logfile, "%s", entry->line);
+
+    *loglist [log_id].log_tail = entry;
+    loglist [log_id].log_tail = &entry->next;
+
+    if (loglist [log_id].entries >= loglist [log_id].keep_entries)
+    {
+        log_entry_t *to_go = loglist [log_id].log_head;
+        loglist [log_id].log_head = to_go->next;
+        loglist [log_id].total -= to_go->len;
+        free (to_go->line);
+        free (to_go);
+    }
+    else
+        loglist [log_id].entries++;
+    return len;
+}
+
+
+void log_contents (int log_id, char **_contents, unsigned int *_len)
+{
+    int remain;
+    log_entry_t *entry;
+    char *ptr;
+
+    if (log_id < 0) return;
+    if (log_id >= LOG_MAXLOGS) return; /* Bad log number */
+
+    _lock_logger ();
+    remain = loglist [log_id].total + 1;
+    *_contents = malloc (remain);
+    *_len = loglist [log_id].total;
+
+    entry = loglist [log_id].log_head;
+    ptr = *_contents;
+    while (entry)
+    {
+        int len = snprintf (ptr, remain, "%s", entry->line);
+        if (len > 0)
+        {
+            ptr += len;
+            remain -= len;
+        }
+        entry = entry->next;
+    }
+    _unlock_logger ();
+}
+
+
 void log_write(int log_id, unsigned priority, const char *cat, const char *func, 
         const char *fmt, ...)
 {
     static char *prior[] = { "EROR", "WARN", "INFO", "DBUG" };
-    char tyme[128];
+    int datelen;
     char filename_tyme[128];
     char pre[256];
     char line[LOG_MAXLINELEN];
     time_t now;
     va_list ap;
 
-    if (log_id < 0) return;
-    if (log_id > LOG_MAXLOGS) return; /* Bad log number */
+    if (log_id < 0 || log_id >= LOG_MAXLOGS) return; /* Bad log number */
     if (loglist[log_id].level < priority) return;
     if (priority > sizeof(prior)/sizeof(prior[0])) return; /* Bad priority */
 
@@ -322,14 +436,14 @@
     now = time(NULL);
 
     _lock_logger();
-    strftime(tyme, sizeof (tyme), "[%Y-%m-%d  %H:%M:%S]", localtime(&now)); 
+    datelen = strftime (pre, sizeof (pre), "[%Y-%m-%d  %H:%M:%S]", localtime(&now)); 
     strftime(filename_tyme, sizeof (filename_tyme), "%Y%m%d_%H%M%S", localtime(&now)); 
 
-    snprintf(pre, sizeof (pre), "%s %s%s", prior[priority-1], cat, func);
+    snprintf (pre+datelen, sizeof (pre)-datelen, " %s %s%s ", prior [priority-1], cat, func);
 
     if (_log_open (log_id, filename_tyme))
     {
-        int len = fprintf (loglist[log_id].logfile, "%s %s %s\n", tyme, pre, line); 
+        int len = create_log_entry (log_id, pre, line);
         if (len > 0)
             loglist[log_id].size += len;
     }
@@ -345,7 +459,7 @@
     char filename_tyme[128];
     time_t now;
 
-    if (log_id < 0) return;
+    if (log_id < 0 || log_id >= LOG_MAXLOGS) return;
     
     va_start(ap, fmt);
 
@@ -356,7 +470,7 @@
     strftime(filename_tyme, sizeof (filename_tyme), "%Y%m%d_%H%M%S", localtime(&now)); 
     if (_log_open (log_id, filename_tyme))
     {
-        int len = fprintf(loglist[log_id].logfile, "%s\n", line);
+        int len = create_log_entry (log_id, "", line);
         if (len > 0)
             loglist[log_id].size += len;
     }
@@ -367,7 +481,7 @@
     fflush(loglist[log_id].logfile);
 }
 
-static int _get_log_id()
+static int _get_log_id(void)
 {
     int i;
     int id = -1;
@@ -399,7 +513,7 @@
     _unlock_logger();
 }
 
-static void _lock_logger()
+static void _lock_logger(void)
 {
 #ifndef _WIN32
     pthread_mutex_lock(&_logger_mutex);
@@ -408,7 +522,7 @@
 #endif
 }
 
-static void _unlock_logger()
+static void _unlock_logger(void)
 {
 #ifndef _WIN32
     pthread_mutex_unlock(&_logger_mutex);
diff -ruN -xMakefile.in icecast2-2.3.1.orig/src/log/log.h icecast/src/log/log.h
--- icecast2-2.3.1.orig/src/log/log.h	2005-10-24 15:43:40.000000000 +0200
+++ icecast/src/log/log.h	2006-06-20 17:29:06.000000000 +0200
@@ -22,18 +22,20 @@
 #define IO_BUFFER_TYPE _IOLBF
 #endif
 
-void log_initialize();
+void log_initialize(void);
 int log_open_file(FILE *file);
 int log_open(const char *filename);
 int log_open_with_buffer(const char *filename, int size);
 void log_set_level(int log_id, unsigned level);
 void log_set_trigger(int id, unsigned trigger);
 int  log_set_filename(int id, const char *filename);
+void log_set_lines_kept (int log_id, unsigned int count);
+void log_contents (int log_id, char **_contents, unsigned int *_len);
 int log_set_archive_timestamp(int id, int value);
 void log_flush(int log_id);
 void log_reopen(int log_id);
 void log_close(int log_id);
-void log_shutdown();
+void log_shutdown(void);
 
 void log_write(int log_id, unsigned priority, const char *cat, const char *func, 
         const char *fmt, ...);
diff -ruN -xMakefile.in icecast2-2.3.1.orig/src/logging.c icecast/src/logging.c
--- icecast2-2.3.1.orig/src/logging.c	2005-10-24 15:47:23.000000000 +0200
+++ icecast/src/logging.c	2006-06-20 17:28:14.000000000 +0200
@@ -25,7 +25,7 @@
 #include "refbuf.h"
 #include "client.h"
 
-#include "os.h"
+#include "compat.h"
 #include "cfgfile.h"
 #include "logging.h"
 #include "util.h"
diff -ruN -xMakefile.in icecast2-2.3.1.orig/src/logging.h icecast/src/logging.h
--- icecast2-2.3.1.orig/src/logging.h	2005-08-20 01:20:56.000000000 +0200
+++ icecast/src/logging.h	2006-06-20 17:28:14.000000000 +0200
@@ -30,33 +30,30 @@
 */
 
 #ifdef _WIN32
-#define __FUNCTION__ strrchr (__FILE__, '\\') ? strrchr (__FILE__, '\\') + 1 : __FILE__
+#include <string.h>
+#define __func__ strrchr (__FILE__, '\\') ? strrchr (__FILE__, '\\') + 1 : __FILE__
 #endif
 
-#ifdef __SUNPRO_C
-#define __FUNCTION__ __func__
-#endif
-
-#define ERROR0(y) log_write(errorlog, 1, CATMODULE "/", __FUNCTION__, y)
-#define ERROR1(y, a) log_write(errorlog, 1, CATMODULE "/", __FUNCTION__, y, a)
-#define ERROR2(y, a, b) log_write(errorlog, 1, CATMODULE "/", __FUNCTION__, y, a, b)
-#define ERROR3(y, a, b, c) log_write(errorlog, 1, CATMODULE "/", __FUNCTION__, y, a, b, c)
-
-#define WARN0(y) log_write(errorlog, 2, CATMODULE "/", __FUNCTION__, y)
-#define WARN1(y, a) log_write(errorlog, 2, CATMODULE "/", __FUNCTION__, y, a)
-#define WARN2(y, a, b) log_write(errorlog, 2, CATMODULE "/", __FUNCTION__, y, a, b)
-#define WARN3(y, a, b, c) log_write(errorlog, 2, CATMODULE "/", __FUNCTION__, y, a, b, c)
-
-#define INFO0(y) log_write(errorlog, 3, CATMODULE "/", __FUNCTION__, y)
-#define INFO1(y, a) log_write(errorlog, 3, CATMODULE "/", __FUNCTION__, y, a)
-#define INFO2(y, a, b) log_write(errorlog, 3, CATMODULE "/", __FUNCTION__, y, a, b)
-#define INFO3(y, a, b, c) log_write(errorlog, 3, CATMODULE "/", __FUNCTION__, y, a, b, c)
-
-#define DEBUG0(y) log_write(errorlog, 4, CATMODULE "/", __FUNCTION__, y)
-#define DEBUG1(y, a) log_write(errorlog, 4, CATMODULE "/", __FUNCTION__, y, a)
-#define DEBUG2(y, a, b) log_write(errorlog, 4, CATMODULE "/", __FUNCTION__, y, a, b)
-#define DEBUG3(y, a, b, c) log_write(errorlog, 4, CATMODULE "/", __FUNCTION__, y, a, b, c)
-#define DEBUG4(y, a, b, c, d) log_write(errorlog, 4, CATMODULE "/", __FUNCTION__, y, a, b, c, d)
+#define ERROR0(y) log_write(errorlog, 1, CATMODULE "/", __func__, y)
+#define ERROR1(y, a) log_write(errorlog, 1, CATMODULE "/", __func__, y, a)
+#define ERROR2(y, a, b) log_write(errorlog, 1, CATMODULE "/", __func__, y, a, b)
+#define ERROR3(y, a, b, c) log_write(errorlog, 1, CATMODULE "/", __func__, y, a, b, c)
+
+#define WARN0(y) log_write(errorlog, 2, CATMODULE "/", __func__, y)
+#define WARN1(y, a) log_write(errorlog, 2, CATMODULE "/", __func__, y, a)
+#define WARN2(y, a, b) log_write(errorlog, 2, CATMODULE "/", __func__, y, a, b)
+#define WARN3(y, a, b, c) log_write(errorlog, 2, CATMODULE "/", __func__, y, a, b, c)
+
+#define INFO0(y) log_write(errorlog, 3, CATMODULE "/", __func__, y)
+#define INFO1(y, a) log_write(errorlog, 3, CATMODULE "/", __func__, y, a)
+#define INFO2(y, a, b) log_write(errorlog, 3, CATMODULE "/", __func__, y, a, b)
+#define INFO3(y, a, b, c) log_write(errorlog, 3, CATMODULE "/", __func__, y, a, b, c)
+
+#define DEBUG0(y) log_write(errorlog, 4, CATMODULE "/", __func__, y)
+#define DEBUG1(y, a) log_write(errorlog, 4, CATMODULE "/", __func__, y, a)
+#define DEBUG2(y, a, b) log_write(errorlog, 4, CATMODULE "/", __func__, y, a, b)
+#define DEBUG3(y, a, b, c) log_write(errorlog, 4, CATMODULE "/", __func__, y, a, b, c)
+#define DEBUG4(y, a, b, c, d) log_write(errorlog, 4, CATMODULE "/", __func__, y, a, b, c, d)
 
 /* CATMODULE is the category or module that logging messages come from.
 ** we set one here in cause someone forgets in the .c file.
diff -ruN -xMakefile.in icecast2-2.3.1.orig/src/main.c icecast/src/main.c
--- icecast2-2.3.1.orig/src/main.c	2005-10-24 16:03:16.000000000 +0200
+++ icecast/src/main.c	2006-06-20 17:28:14.000000000 +0200
@@ -39,7 +39,7 @@
 #include "sighandler.h"
 
 #include "global.h"
-#include "os.h"
+#include "compat.h"
 #include "connection.h"
 #include "refbuf.h"
 #include "client.h"
@@ -75,9 +75,9 @@
 #endif
 }
 
-static void _print_usage()
+static void _print_usage(void)
 {
-    printf(ICECAST_VERSION_STRING "\n\n");
+    printf("%s\n\n", ICECAST_VERSION_STRING);
     printf("usage: icecast [-b -v] -c <file>\n");
     printf("options:\n");
     printf("\t-c <file>\tSpecify configuration file\n");
@@ -518,7 +518,7 @@
         return 1;
     }
 
-    INFO0 (ICECAST_VERSION_STRING " server started");
+    INFO1 ("%s server started", ICECAST_VERSION_STRING);
 
     /* REM 3D Graphics */
 
diff -ruN -xMakefile.in icecast2-2.3.1.orig/src/Makefile.am icecast/src/Makefile.am
--- icecast2-2.3.1.orig/src/Makefile.am	2005-08-19 04:01:58.000000000 +0200
+++ icecast/src/Makefile.am	2006-06-20 17:28:14.000000000 +0200
@@ -6,7 +6,7 @@
 
 bin_PROGRAMS = icecast
 
-noinst_HEADERS = admin.h cfgfile.h os.h logging.h sighandler.h connection.h \
+noinst_HEADERS = admin.h cfgfile.h logging.h sighandler.h connection.h \
     global.h util.h slave.h source.h stats.h refbuf.h client.h \
     compat.h fserve.h xslt.h yp.h event.h md5.h \
     auth.h auth_htpasswd.h auth_url.h \
diff -ruN -xMakefile.in icecast2-2.3.1.orig/src/net/sock.h icecast/src/net/sock.h
--- icecast2-2.3.1.orig/src/net/sock.h	2005-08-19 04:01:59.000000000 +0200
+++ icecast/src/net/sock.h	2006-06-20 17:29:18.000000000 +0200
@@ -31,7 +31,7 @@
 #ifdef HAVE_UNISTD_H
 #include <unistd.h>
 #elif _WIN32
-#include <os.h>
+#include <compat.h>
 #endif
 
 #ifdef HAVE_SYS_UIO_H
diff -ruN -xMakefile.in icecast2-2.3.1.orig/src/os.h icecast/src/os.h
--- icecast2-2.3.1.orig/src/os.h	2005-08-19 04:01:58.000000000 +0200
+++ icecast/src/os.h	1970-01-01 01:00:00.000000000 +0100
@@ -1,30 +0,0 @@
-/* Icecast
- *
- * This program is distributed under the GNU General Public License, version 2.
- * A copy of this license is included with this source.
- *
- * Copyright 2000-2004, Jack Moffitt <jack@xiph.org, 
- *                      Michael Smith <msmith@xiph.org>,
- *                      oddsock <oddsock@xiph.org>,
- *                      Karl Heyes <karl@xiph.org>
- *                      and others (see AUTHORS for details).
- */
-
-#ifndef __OS_H__
-#define __OS_H__
-
-#ifdef _WIN32
-#include <windows.h>
-#else
-#include <unistd.h>
-#endif
-
-#ifdef _WIN32
-#define PATH_SEPARATOR "\\"
-#define size_t int
-#define ssize_t int
-#else
-#define PATH_SEPARATOR "/"
-#endif
-
-#endif  /* __OS_H__ */
diff -ruN -xMakefile.in icecast2-2.3.1.orig/src/slave.c icecast/src/slave.c
--- icecast2-2.3.1.orig/src/slave.c	2005-09-08 16:03:51.000000000 +0200
+++ icecast/src/slave.c	2006-06-20 17:28:14.000000000 +0200
@@ -39,7 +39,7 @@
 #define strncasecmp strnicmp
 #endif
 
-#include "os.h"
+#include "compat.h"
 
 #include "thread/thread.h"
 #include "avl/avl.h"
@@ -65,7 +65,6 @@
 static int slave_running = 0;
 static int update_settings = 0;
 static volatile unsigned int max_interval = 0;
-static volatile int rescan_relays = 0;
 
 relay_server *relay_free (relay_server *relay)
 {
@@ -116,22 +115,12 @@
 }
 
 
-/* Request slave thread to rescan the existing relays to see if any need
- * starting up, eg on-demand relays
- */
-void slave_rescan (void)
-{
-    rescan_relays = 1;
-}
-
-
 /* Request slave thread to check the relay list for changes and to
  * update the stats for the current streams.
  */
 void slave_rebuild_mounts (void)
 {
     update_settings = 1;
-    rescan_relays = 1;
 }
 
 
@@ -209,11 +198,12 @@
          * we don't send in this header then relay will not have mp3 metadata.
          */
         sock_write(streamsock, "GET %s HTTP/1.0\r\n"
-                "User-Agent: " ICECAST_VERSION_STRING "\r\n"
+                "User-Agent: %s\r\n"
                 "%s"
                 "%s"
                 "\r\n",
                 relay->mount,
+                ICECAST_VERSION_STRING,
                 relay->mp3metadata?"Icy-MetaData: 1\r\n":"",
                 auth_header);
         free (auth_header);
@@ -248,6 +238,7 @@
             break;
         }
         global_unlock ();
+        sock_set_blocking (streamsock, SOCK_NONBLOCK);
         con = NULL;
         parser = NULL;
         client_set_queue (src->client, NULL);
@@ -267,11 +258,11 @@
             /* only keep refreshing YP entries for inactive on-demand relays */
             yp_remove (relay->localmount);
             relay->source->yp_public = -1;
+            relay->start = time(NULL) + 10; /* prevent busy looping if failing */
         }
 
-        /* initiate an immediate relay cleanup run */
+        /* we've finished, now get cleaned up */
         relay->cleanup = 1;
-        rescan_relays = 1;
 
         return NULL;
     } while (0);
@@ -298,9 +289,9 @@
     src->parser = NULL;
     source_clear_source (relay->source);
 
-    /* initiate an immediate relay cleanup run */
+    /* cleanup relay, but prevent this relay from starting up again too soon */
+    relay->start = time(NULL) + max_interval;
     relay->cleanup = 1;
-    rescan_relays = 1;
 
     return NULL;
 }
@@ -320,25 +311,21 @@
         /* new relay, reserve the name */
         relay->source = source_reserve (relay->localmount);
         if (relay->source)
+        {
             DEBUG1("Adding relay source at mountpoint \"%s\"", relay->localmount);
+            slave_rebuild_mounts();
+        }
         else
             WARN1 ("new relay but source \"%s\" already exists", relay->localmount);
     }
     do
     {
         source_t *source = relay->source;
-        if (relay->source == NULL || relay->running)
+        /* skip relay if active, not configured or just not time yet */
+        if (relay->source == NULL || relay->running || relay->start > time(NULL))
             break;
-        if (relay->on_demand)
+        if (relay->on_demand && source->on_demand_req == 0)
         {
-            ice_config_t *config = config_get_config ();
-            mount_proxy *mountinfo = config_find_mount (config, relay->localmount);
-
-            if (mountinfo == NULL)
-                source_update_settings (config, relay->source, mountinfo);
-            config_release_config ();
-            slave_rebuild_mounts();
-            stats_event (relay->localmount, "listeners", "0");
             relay->source->on_demand = relay->on_demand;
 
             if (source->fallback_mount && source->fallback_override)
@@ -358,17 +345,21 @@
                 break;
         }
 
+        relay->start = time(NULL) + 5;
         relay->thread = thread_create ("Relay Thread", start_relay_stream,
                 relay, THREAD_ATTACHED);
         return;
 
-    } while(0);
+    } while (0);
     /* the relay thread may of shut down itself */
-    if (relay->cleanup && relay->thread)
+    if (relay->cleanup)
     {
-        DEBUG1 ("waiting for relay thread for \"%s\"", relay->localmount);
-        thread_join (relay->thread);
-        relay->thread = NULL;
+        if (relay->thread)
+        {
+            DEBUG1 ("waiting for relay thread for \"%s\"", relay->localmount);
+            thread_join (relay->thread);
+            relay->thread = NULL;
+        }
         relay->cleanup = 0;
         relay->running = 0;
 
@@ -468,7 +459,8 @@
 }
 
 
-static void relay_check_streams (relay_server *to_start, relay_server *to_free)
+static void relay_check_streams (relay_server *to_start,
+        relay_server *to_free, int skip_timer)
 {
     relay_server *relay;
 
@@ -493,6 +485,8 @@
     relay = to_start;
     while (relay)
     {
+        if (skip_timer)
+            relay->start = 0;
         check_relay_stream (relay);
         relay = relay->next;
     }
@@ -583,8 +577,8 @@
         thread_mutex_lock (&(config_locks()->relay_lock));
         cleanup_relays = update_relays (&global.master_relays, new_relays);
         
-        relay_check_streams (global.master_relays, cleanup_relays);
-        relay_check_streams (NULL, new_relays);
+        relay_check_streams (global.master_relays, cleanup_relays, 0);
+        relay_check_streams (NULL, new_relays, 0);
 
         thread_mutex_unlock (&(config_locks()->relay_lock));
 
@@ -610,7 +604,8 @@
 
     while (1)
     {
-        relay_server *cleanup_relays;
+        relay_server *cleanup_relays = NULL;
+        int skip_timer = 0;
 
         /* re-read xml file if requested */
         if (global . schedule_config_reread)
@@ -622,8 +617,8 @@
         thread_sleep (1000000);
         if (slave_running == 0)
             break;
-        if (rescan_relays == 0 && max_interval > ++interval)
-            continue;
+
+        ++interval;
 
         /* only update relays lists when required */
         if (max_interval <= interval)
@@ -631,6 +626,8 @@
             DEBUG0 ("checking master stream list");
             config = config_get_config();
 
+            if (max_interval == 0)
+                skip_timer = 1;
             interval = 0;
             max_interval = config->master_update_interval;
 
@@ -643,19 +640,14 @@
             cleanup_relays = update_relays (&global.relays, config->relay);
 
             config_release_config();
-
-            relay_check_streams (global.relays, cleanup_relays);
-            thread_mutex_unlock (&(config_locks()->relay_lock));
         }
         else
-        {
-            DEBUG0 ("rescanning relay lists");
             thread_mutex_lock (&(config_locks()->relay_lock));
-            relay_check_streams (global.master_relays, NULL);
-            relay_check_streams (global.relays, NULL);
-            thread_mutex_unlock (&(config_locks()->relay_lock));
-        }
-        rescan_relays = 0;
+
+        relay_check_streams (global.relays, cleanup_relays, skip_timer);
+        relay_check_streams (global.master_relays, NULL, skip_timer);
+        thread_mutex_unlock (&(config_locks()->relay_lock));
+
         if (update_settings)
         {
             update_settings = 0;
@@ -663,8 +655,8 @@
         }
     }
     DEBUG0 ("shutting down current relays");
-    relay_check_streams (NULL, global.relays);
-    relay_check_streams (NULL, global.master_relays);
+    relay_check_streams (NULL, global.relays, 0);
+    relay_check_streams (NULL, global.master_relays, 0);
 
     INFO0 ("Slave thread shutdown complete");
 
diff -ruN -xMakefile.in icecast2-2.3.1.orig/src/slave.h icecast/src/slave.h
--- icecast2-2.3.1.orig/src/slave.h	2005-08-19 04:01:58.000000000 +0200
+++ icecast/src/slave.h	2006-06-20 17:28:14.000000000 +0200
@@ -27,6 +27,7 @@
     int on_demand;
     int running;
     int cleanup;
+    time_t start;
     thread_type *thread;
     struct _relay_server *next;
 } relay_server;
@@ -36,7 +37,6 @@
 void slave_shutdown(void);
 void slave_recheck_mounts (void);
 void slave_rebuild_mounts (void);
-void slave_rescan (void);
 relay_server *relay_free (relay_server *relay);
 
 #endif  /* __SLAVE_H__ */
diff -ruN -xMakefile.in icecast2-2.3.1.orig/src/source.c icecast/src/source.c
--- icecast2-2.3.1.orig/src/source.c	2005-11-30 19:16:17.000000000 +0100
+++ icecast/src/source.c	2006-06-20 17:28:14.000000000 +0200
@@ -50,7 +50,7 @@
 #include "format.h"
 #include "fserve.h"
 #include "auth.h"
-#include "os.h"
+#include "compat.h"
 
 #undef CATMODULE
 #define CATMODULE "source"
@@ -194,11 +194,17 @@
 void source_clear_source (source_t *source)
 {
     DEBUG1 ("clearing source \"%s\"", source->mount);
+
+    avl_tree_wlock (source->client_tree);
     client_destroy(source->client);
     source->client = NULL;
     source->parser = NULL;
     source->con = NULL;
 
+    /* log bytes read in access log */
+    if (source->client && source->format)
+        source->client->con->sent_bytes = source->format->read_bytes;
+
     if (source->dumpfile)
     {
         INFO1 ("Closing dumpfile for %s", source->mount);
@@ -207,15 +213,13 @@
     }
 
     /* lets kick off any clients that are left on here */
-    avl_tree_rlock (source->client_tree);
     while (avl_get_first (source->client_tree))
     {
         avl_delete (source->client_tree,
                 avl_get_first (source->client_tree)->key, _free_client);
     }
-    avl_tree_unlock (source->client_tree);
 
-    avl_tree_rlock (source->pending_tree);
+    avl_tree_wlock (source->pending_tree);
     while (avl_get_first (source->pending_tree))
     {
         avl_delete (source->pending_tree,
@@ -265,6 +269,7 @@
     }
 
     source->on_demand_req = 0;
+    avl_tree_unlock (source->client_tree);
 }
 
 
@@ -585,9 +590,7 @@
 
     stats_event (source->mount, "listenurl", listenurl);
 
-    if (listenurl) {
-        free(listenurl);
-    }
+    free(listenurl);
 
     if (source->dumpfilename != NULL)
     {
@@ -933,6 +936,7 @@
     http_parser_t *parser = NULL;
 
     DEBUG1("Applying mount information for \"%s\"", source->mount);
+    avl_tree_rlock (source->client_tree);
     if (mountinfo)
     {
         source->max_listeners = mountinfo->max_listeners;
@@ -1131,6 +1135,7 @@
 
     if (source->format && source->format->apply_settings)
         source->format->apply_settings (source->client, source->format, mountinfo);
+    avl_tree_unlock (source->client_tree);
 }
 
 
diff -ruN -xMakefile.in icecast2-2.3.1.orig/src/stats.c icecast/src/stats.c
--- icecast2-2.3.1.orig/src/stats.c	2005-11-30 19:16:17.000000000 +0100
+++ icecast/src/stats.c	2006-06-20 17:28:14.000000000 +0200
@@ -119,7 +119,7 @@
     thread_mutex_unlock(&_global_event_mutex);
 }
 
-void stats_initialize()
+void stats_initialize(void)
 {
     _event_listeners = NULL;
 
@@ -139,7 +139,7 @@
     _stats_thread_id = thread_create("Stats Thread", _stats_thread, NULL, THREAD_ATTACHED);
 }
 
-void stats_shutdown()
+void stats_shutdown(void)
 {
     int n;
 
@@ -182,7 +182,7 @@
     }
 }
 
-stats_t *stats_get_stats()
+stats_t *stats_get_stats(void)
 {
     /* lock global stats
     
diff -ruN -xMakefile.in icecast2-2.3.1.orig/src/stats.h icecast/src/stats.h
--- icecast2-2.3.1.orig/src/stats.h	2005-08-19 04:01:58.000000000 +0200
+++ icecast/src/stats.h	2006-06-20 17:28:14.000000000 +0200
@@ -70,10 +70,10 @@
 
 } stats_t;
 
-void stats_initialize();
-void stats_shutdown();
+void stats_initialize(void);
+void stats_shutdown(void);
 
-stats_t *stats_get_stats();
+stats_t *stats_get_stats(void);
 
 void stats_event(const char *source, const char *name, const char *value);
 void stats_event_args(const char *source, char *name, char *format, ...);
diff -ruN -xMakefile.in icecast2-2.3.1.orig/src/util.c icecast/src/util.c
--- icecast2-2.3.1.orig/src/util.c	2005-08-19 04:01:58.000000000 +0200
+++ icecast/src/util.c	2006-06-20 17:28:14.000000000 +0200
@@ -40,7 +40,7 @@
 
 #include "cfgfile.h"
 #include "util.h"
-#include "os.h"
+#include "compat.h"
 #include "refbuf.h"
 #include "connection.h"
 #include "client.h"
diff -ruN -xMakefile.in icecast2-2.3.1.orig/src/xslt.c icecast/src/xslt.c
--- icecast2-2.3.1.orig/src/xslt.c	2005-08-20 01:20:56.000000000 +0200
+++ icecast/src/xslt.c	2006-06-20 17:28:14.000000000 +0200
@@ -94,7 +94,7 @@
 static stylesheet_cache_t cache[CACHESIZE];
 static mutex_t xsltlock;
 
-void xslt_initialize()
+void xslt_initialize(void)
 {
     xmlSubstituteEntitiesDefault(1);
     xmlLoadExtDtdDefaultValue = 1;
@@ -105,7 +105,7 @@
     xmlLoadExtDtdDefaultValue = 1;
 }
 
-void xslt_shutdown() {
+void xslt_shutdown(void) {
     int i;
 
     for(i=0; i < CACHESIZE; i++) {
@@ -119,7 +119,7 @@
     xsltCleanupGlobals();
 }
 
-static int evict_cache_entry() {
+static int evict_cache_entry(void) {
     int i, age=0, oldest=0;
 
     for(i=0; i < CACHESIZE; i++) {
diff -ruN -xMakefile.in icecast2-2.3.1.orig/src/xslt.h icecast/src/xslt.h
--- icecast2-2.3.1.orig/src/xslt.h	2005-08-19 04:01:58.000000000 +0200
+++ icecast/src/xslt.h	2006-06-20 17:28:14.000000000 +0200
@@ -35,6 +35,6 @@
 
 
 void xslt_transform(xmlDocPtr doc, const char *xslfilename, client_t *client);
-void xslt_initialize();
-void xslt_shutdown();
+void xslt_initialize(void);
+void xslt_shutdown(void);
 
diff -ruN -xMakefile.in icecast2-2.3.1.orig/src/yp.c icecast/src/yp.c
--- icecast2-2.3.1.orig/src/yp.c	2005-09-08 16:03:51.000000000 +0200
+++ icecast/src/yp.c	2006-06-20 17:28:14.000000000 +0200
@@ -265,7 +265,7 @@
 }
 
 
-void yp_initialize()
+void yp_initialize(void)
 {
     ice_config_t *config = config_get_config();
     thread_rwlock_create (&yp_lock);
@@ -562,7 +562,7 @@
 
 
 /* Check for changes in the YP servers configured */
-static void check_servers ()
+static void check_servers (void)
 {
     struct yp_server *server = (struct yp_server *)active_yps,
                      **server_p = (struct yp_server **)&active_yps;
@@ -812,6 +812,8 @@
             free (yp->subtype);
             yp->subtype = escaped;
             break;
+        default:
+            free (escaped);
     }
 }
 
@@ -916,7 +918,7 @@
 }
 
 
-void yp_shutdown ()
+void yp_shutdown (void)
 {
     yp_running = 0;
     yp_update = 1;
diff -ruN -xMakefile.in icecast2-2.3.1.orig/src/yp.h icecast/src/yp.h
--- icecast2-2.3.1.orig/src/yp.h	2005-08-19 04:01:58.000000000 +0200
+++ icecast/src/yp.h	2006-06-20 17:28:14.000000000 +0200
@@ -32,8 +32,8 @@
 void yp_remove (const char *mount);
 void yp_touch (const char *mount);
 void yp_recheck_config (ice_config_t *config);
-void yp_initialize();
-void yp_shutdown();
+void yp_initialize(void);
+void yp_shutdown(void);
 
 #else
 
