Index: icecast2-2.3.2/Makefile.am
===================================================================
--- icecast2-2.3.2.orig/Makefile.am	2007-11-20 03:38:17.000000000 +0100
+++ icecast2-2.3.2/Makefile.am	2009-07-31 03:01:27.000000000 +0200
@@ -3,7 +3,7 @@
 AUTOMAKE_OPTIONS = foreign dist-zip
 ACLOCAL_AMFLAGS = -I m4
 
-SUBDIRS = src conf debian doc web admin win32 examples
+SUBDIRS = src conf doc web admin win32 examples
 
 EXTRA_DIST = HACKING m4/acx_pthread.m4 m4/ogg.m4 \
     m4/theora.m4 m4/vorbis.m4 m4/speex.m4 \
Index: icecast2-2.3.2/configure.in
===================================================================
--- icecast2-2.3.2.orig/configure.in	2008-05-27 02:21:38.000000000 +0200
+++ icecast2-2.3.2/configure.in	2009-07-31 03:01:27.000000000 +0200
@@ -150,7 +150,7 @@
 AC_SUBST(HAVE_KATE)
 AC_SUBST(KATE_LIBS)
 
-AC_OUTPUT([Makefile conf/Makefile debian/Makefile src/Makefile src/avl/Makefile
+AC_OUTPUT([Makefile conf/Makefile src/Makefile src/avl/Makefile
 src/httpp/Makefile src/thread/Makefile src/log/Makefile
 src/net/Makefile src/timing/Makefile doc/Makefile web/Makefile
 admin/Makefile win32/Makefile win32/res/Makefile examples/Makefile])
