--- 1/icecast2-2.3.1/src/admin.c	2005-09-13 01:50:48.000000000 +0300
+++ icecast2-2.3.1/src/admin.c	2007-01-03 15:07:18.000000000 +0200
@@ -974,34 +974,32 @@
         char *buf;
         int remaining = PER_CLIENT_REFBUF_SIZE;
         int ret;
-        ice_config_t *config = config_get_config ();
-        mount_proxy *mountinfo = config->mounts;
+	avl_node *node;
+        source_t *source;
+
 
         buf = client->refbuf->data;
         ret = snprintf (buf, remaining,
                 "HTTP/1.0 200 OK\r\nContent-Type: text/html\r\n\r\n");
 
-        while (mountinfo && ret > 0 && ret < remaining)
+	node = avl_get_first(global.source_tree);
+        while (node && ret > 0 && ret < remaining)
         {
-            mount_proxy *current = mountinfo;
-            source_t *source;
-            mountinfo = mountinfo->next;
 
-            /* now check that a source is available */
-            source = source_find_mount (current->mountname);
+	    source = (source_t *)node->key;
+	    node = avl_get_next(node);
 
             if (source == NULL)
                 continue;
-            if (source->running == 0 && source->on_demand == 0)
+	    if (source->running == 0 && source->on_demand == 0)
                 continue;
             if (source->hidden)
                 continue;
             remaining -= ret;
             buf += ret;
-            ret = snprintf (buf, remaining, "%s\n", current->mountname);
+            ret = snprintf (buf, remaining, "%s\n", source->mount);
         }
         avl_tree_unlock (global.source_tree);
-        config_release_config();
 
         /* handle last line */
         if (ret > 0 && ret < remaining)
