Index: icecast2-2.3.2/src/stats.c
===================================================================
--- icecast2-2.3.2.orig/src/stats.c	2009-12-09 09:53:27.000000000 -0600
+++ icecast2-2.3.2/src/stats.c	2009-12-09 09:54:16.000000000 -0600
@@ -973,7 +973,11 @@
                 if (event->source)
                 {
                     if (show_mount && strcmp (event->source, show_mount) != 0)
+                    {
+                        xmlFree (value);
+                        xmlFree (name);
                         break;
+                    }
                     srcnode = _find_xml_node(event->source, &src_nodes, node);
                 }
                 else
